// @aws-sdk/client-sesv2/dist-es/models/models_0.js
import { __assign, __extends } from "tslib";
import { SENSITIVE_STRING } from "@aws-sdk/smithy-client";
import { SESv2ServiceException as __BaseException } from "./SESv2ServiceException";
export var ContactLanguage;
(function (ContactLanguage) {
    ContactLanguage["EN"] = "EN";
    ContactLanguage["JA"] = "JA";
})(ContactLanguage || (ContactLanguage = {}));
export var MailType;
(function (MailType) {
    MailType["MARKETING"] = "MARKETING";
    MailType["TRANSACTIONAL"] = "TRANSACTIONAL";
})(MailType || (MailType = {}));
export var ReviewStatus;
(function (ReviewStatus) {
    ReviewStatus["DENIED"] = "DENIED";
    ReviewStatus["FAILED"] = "FAILED";
    ReviewStatus["GRANTED"] = "GRANTED";
    ReviewStatus["PENDING"] = "PENDING";
})(ReviewStatus || (ReviewStatus = {}));
var AccountSuspendedException = (function (_super) {
    __extends(AccountSuspendedException, _super);
    function AccountSuspendedException(opts) {
        var _this = _super.call(this, __assign({ name: "AccountSuspendedException", $fault: "client" }, opts)) || this;
        _this.name = "AccountSuspendedException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, AccountSuspendedException.prototype);
        return _this;
    }
    return AccountSuspendedException;
}(__BaseException));
export { AccountSuspendedException };
var AlreadyExistsException = (function (_super) {
    __extends(AlreadyExistsException, _super);
    function AlreadyExistsException(opts) {
        var _this = _super.call(this, __assign({ name: "AlreadyExistsException", $fault: "client" }, opts)) || this;
        _this.name = "AlreadyExistsException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, AlreadyExistsException.prototype);
        return _this;
    }
    return AlreadyExistsException;
}(__BaseException));
export { AlreadyExistsException };
var BadRequestException = (function (_super) {
    __extends(BadRequestException, _super);
    function BadRequestException(opts) {
        var _this = _super.call(this, __assign({ name: "BadRequestException", $fault: "client" }, opts)) || this;
        _this.name = "BadRequestException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, BadRequestException.prototype);
        return _this;
    }
    return BadRequestException;
}(__BaseException));
export { BadRequestException };
export var BehaviorOnMxFailure;
(function (BehaviorOnMxFailure) {
    BehaviorOnMxFailure["REJECT_MESSAGE"] = "REJECT_MESSAGE";
    BehaviorOnMxFailure["USE_DEFAULT_VALUE"] = "USE_DEFAULT_VALUE";
})(BehaviorOnMxFailure || (BehaviorOnMxFailure = {}));
export var BulkEmailStatus;
(function (BulkEmailStatus) {
    BulkEmailStatus["ACCOUNT_DAILY_QUOTA_EXCEEDED"] = "ACCOUNT_DAILY_QUOTA_EXCEEDED";
    BulkEmailStatus["ACCOUNT_SENDING_PAUSED"] = "ACCOUNT_SENDING_PAUSED";
    BulkEmailStatus["ACCOUNT_SUSPENDED"] = "ACCOUNT_SUSPENDED";
    BulkEmailStatus["ACCOUNT_THROTTLED"] = "ACCOUNT_THROTTLED";
    BulkEmailStatus["CONFIGURATION_SET_NOT_FOUND"] = "CONFIGURATION_SET_NOT_FOUND";
    BulkEmailStatus["CONFIGURATION_SET_SENDING_PAUSED"] = "CONFIGURATION_SET_SENDING_PAUSED";
    BulkEmailStatus["FAILED"] = "FAILED";
    BulkEmailStatus["INVALID_PARAMETER"] = "INVALID_PARAMETER";
    BulkEmailStatus["INVALID_SENDING_POOL_NAME"] = "INVALID_SENDING_POOL_NAME";
    BulkEmailStatus["MAIL_FROM_DOMAIN_NOT_VERIFIED"] = "MAIL_FROM_DOMAIN_NOT_VERIFIED";
    BulkEmailStatus["MESSAGE_REJECTED"] = "MESSAGE_REJECTED";
    BulkEmailStatus["SUCCESS"] = "SUCCESS";
    BulkEmailStatus["TEMPLATE_NOT_FOUND"] = "TEMPLATE_NOT_FOUND";
    BulkEmailStatus["TRANSIENT_FAILURE"] = "TRANSIENT_FAILURE";
})(BulkEmailStatus || (BulkEmailStatus = {}));
export var DimensionValueSource;
(function (DimensionValueSource) {
    DimensionValueSource["EMAIL_HEADER"] = "EMAIL_HEADER";
    DimensionValueSource["LINK_TAG"] = "LINK_TAG";
    DimensionValueSource["MESSAGE_TAG"] = "MESSAGE_TAG";
})(DimensionValueSource || (DimensionValueSource = {}));
var ConcurrentModificationException = (function (_super) {
    __extends(ConcurrentModificationException, _super);
    function ConcurrentModificationException(opts) {
        var _this = _super.call(this, __assign({ name: "ConcurrentModificationException", $fault: "server" }, opts)) || this;
        _this.name = "ConcurrentModificationException";
        _this.$fault = "server";
        Object.setPrototypeOf(_this, ConcurrentModificationException.prototype);
        return _this;
    }
    return ConcurrentModificationException;
}(__BaseException));
export { ConcurrentModificationException };
var ConflictException = (function (_super) {
    __extends(ConflictException, _super);
    function ConflictException(opts) {
        var _this = _super.call(this, __assign({ name: "ConflictException", $fault: "client" }, opts)) || this;
        _this.name = "ConflictException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, ConflictException.prototype);
        return _this;
    }
    return ConflictException;
}(__BaseException));
export { ConflictException };
export var SubscriptionStatus;
(function (SubscriptionStatus) {
    SubscriptionStatus["OPT_IN"] = "OPT_IN";
    SubscriptionStatus["OPT_OUT"] = "OPT_OUT";
})(SubscriptionStatus || (SubscriptionStatus = {}));
export var ContactListImportAction;
(function (ContactListImportAction) {
    ContactListImportAction["DELETE"] = "DELETE";
    ContactListImportAction["PUT"] = "PUT";
})(ContactListImportAction || (ContactListImportAction = {}));
export var TlsPolicy;
(function (TlsPolicy) {
    TlsPolicy["OPTIONAL"] = "OPTIONAL";
    TlsPolicy["REQUIRE"] = "REQUIRE";
})(TlsPolicy || (TlsPolicy = {}));
export var SuppressionListReason;
(function (SuppressionListReason) {
    SuppressionListReason["BOUNCE"] = "BOUNCE";
    SuppressionListReason["COMPLAINT"] = "COMPLAINT";
})(SuppressionListReason || (SuppressionListReason = {}));
var LimitExceededException = (function (_super) {
    __extends(LimitExceededException, _super);
    function LimitExceededException(opts) {
        var _this = _super.call(this, __assign({ name: "LimitExceededException", $fault: "client" }, opts)) || this;
        _this.name = "LimitExceededException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, LimitExceededException.prototype);
        return _this;
    }
    return LimitExceededException;
}(__BaseException));
export { LimitExceededException };
var NotFoundException = (function (_super) {
    __extends(NotFoundException, _super);
    function NotFoundException(opts) {
        var _this = _super.call(this, __assign({ name: "NotFoundException", $fault: "client" }, opts)) || this;
        _this.name = "NotFoundException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, NotFoundException.prototype);
        return _this;
    }
    return NotFoundException;
}(__BaseException));
export { NotFoundException };
var TooManyRequestsException = (function (_super) {
    __extends(TooManyRequestsException, _super);
    function TooManyRequestsException(opts) {
        var _this = _super.call(this, __assign({ name: "TooManyRequestsException", $fault: "client" }, opts)) || this;
        _this.name = "TooManyRequestsException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, TooManyRequestsException.prototype);
        return _this;
    }
    return TooManyRequestsException;
}(__BaseException));
export { TooManyRequestsException };
export var EventType;
(function (EventType) {
    EventType["BOUNCE"] = "BOUNCE";
    EventType["CLICK"] = "CLICK";
    EventType["COMPLAINT"] = "COMPLAINT";
    EventType["DELIVERY"] = "DELIVERY";
    EventType["DELIVERY_DELAY"] = "DELIVERY_DELAY";
    EventType["OPEN"] = "OPEN";
    EventType["REJECT"] = "REJECT";
    EventType["RENDERING_FAILURE"] = "RENDERING_FAILURE";
    EventType["SEND"] = "SEND";
    EventType["SUBSCRIPTION"] = "SUBSCRIPTION";
})(EventType || (EventType = {}));
export var DeliverabilityTestStatus;
(function (DeliverabilityTestStatus) {
    DeliverabilityTestStatus["COMPLETED"] = "COMPLETED";
    DeliverabilityTestStatus["IN_PROGRESS"] = "IN_PROGRESS";
})(DeliverabilityTestStatus || (DeliverabilityTestStatus = {}));
var MailFromDomainNotVerifiedException = (function (_super) {
    __extends(MailFromDomainNotVerifiedException, _super);
    function MailFromDomainNotVerifiedException(opts) {
        var _this = _super.call(this, __assign({ name: "MailFromDomainNotVerifiedException", $fault: "client" }, opts)) || this;
        _this.name = "MailFromDomainNotVerifiedException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, MailFromDomainNotVerifiedException.prototype);
        return _this;
    }
    return MailFromDomainNotVerifiedException;
}(__BaseException));
export { MailFromDomainNotVerifiedException };
var MessageRejected = (function (_super) {
    __extends(MessageRejected, _super);
    function MessageRejected(opts) {
        var _this = _super.call(this, __assign({ name: "MessageRejected", $fault: "client" }, opts)) || this;
        _this.name = "MessageRejected";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, MessageRejected.prototype);
        return _this;
    }
    return MessageRejected;
}(__BaseException));
export { MessageRejected };
var SendingPausedException = (function (_super) {
    __extends(SendingPausedException, _super);
    function SendingPausedException(opts) {
        var _this = _super.call(this, __assign({ name: "SendingPausedException", $fault: "client" }, opts)) || this;
        _this.name = "SendingPausedException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, SendingPausedException.prototype);
        return _this;
    }
    return SendingPausedException;
}(__BaseException));
export { SendingPausedException };
export var DkimSigningKeyLength;
(function (DkimSigningKeyLength) {
    DkimSigningKeyLength["RSA_1024_BIT"] = "RSA_1024_BIT";
    DkimSigningKeyLength["RSA_2048_BIT"] = "RSA_2048_BIT";
})(DkimSigningKeyLength || (DkimSigningKeyLength = {}));
export var DkimSigningAttributesOrigin;
(function (DkimSigningAttributesOrigin) {
    DkimSigningAttributesOrigin["AWS_SES"] = "AWS_SES";
    DkimSigningAttributesOrigin["EXTERNAL"] = "EXTERNAL";
})(DkimSigningAttributesOrigin || (DkimSigningAttributesOrigin = {}));
export var DkimStatus;
(function (DkimStatus) {
    DkimStatus["FAILED"] = "FAILED";
    DkimStatus["NOT_STARTED"] = "NOT_STARTED";
    DkimStatus["PENDING"] = "PENDING";
    DkimStatus["SUCCESS"] = "SUCCESS";
    DkimStatus["TEMPORARY_FAILURE"] = "TEMPORARY_FAILURE";
})(DkimStatus || (DkimStatus = {}));
export var IdentityType;
(function (IdentityType) {
    IdentityType["DOMAIN"] = "DOMAIN";
    IdentityType["EMAIL_ADDRESS"] = "EMAIL_ADDRESS";
    IdentityType["MANAGED_DOMAIN"] = "MANAGED_DOMAIN";
})(IdentityType || (IdentityType = {}));
export var DataFormat;
(function (DataFormat) {
    DataFormat["CSV"] = "CSV";
    DataFormat["JSON"] = "JSON";
})(DataFormat || (DataFormat = {}));
export var SuppressionListImportAction;
(function (SuppressionListImportAction) {
    SuppressionListImportAction["DELETE"] = "DELETE";
    SuppressionListImportAction["PUT"] = "PUT";
})(SuppressionListImportAction || (SuppressionListImportAction = {}));
export var WarmupStatus;
(function (WarmupStatus) {
    WarmupStatus["DONE"] = "DONE";
    WarmupStatus["IN_PROGRESS"] = "IN_PROGRESS";
})(WarmupStatus || (WarmupStatus = {}));
export var DeliverabilityDashboardAccountStatus;
(function (DeliverabilityDashboardAccountStatus) {
    DeliverabilityDashboardAccountStatus["ACTIVE"] = "ACTIVE";
    DeliverabilityDashboardAccountStatus["DISABLED"] = "DISABLED";
    DeliverabilityDashboardAccountStatus["PENDING_EXPIRATION"] = "PENDING_EXPIRATION";
})(DeliverabilityDashboardAccountStatus || (DeliverabilityDashboardAccountStatus = {}));
export var MailFromDomainStatus;
(function (MailFromDomainStatus) {
    MailFromDomainStatus["FAILED"] = "FAILED";
    MailFromDomainStatus["PENDING"] = "PENDING";
    MailFromDomainStatus["SUCCESS"] = "SUCCESS";
    MailFromDomainStatus["TEMPORARY_FAILURE"] = "TEMPORARY_FAILURE";
})(MailFromDomainStatus || (MailFromDomainStatus = {}));
export var JobStatus;
(function (JobStatus) {
    JobStatus["COMPLETED"] = "COMPLETED";
    JobStatus["CREATED"] = "CREATED";
    JobStatus["FAILED"] = "FAILED";
    JobStatus["PROCESSING"] = "PROCESSING";
})(JobStatus || (JobStatus = {}));
export var ImportDestinationType;
(function (ImportDestinationType) {
    ImportDestinationType["CONTACT_LIST"] = "CONTACT_LIST";
    ImportDestinationType["SUPPRESSION_LIST"] = "SUPPRESSION_LIST";
})(ImportDestinationType || (ImportDestinationType = {}));
var InvalidNextTokenException = (function (_super) {
    __extends(InvalidNextTokenException, _super);
    function InvalidNextTokenException(opts) {
        var _this = _super.call(this, __assign({ name: "InvalidNextTokenException", $fault: "client" }, opts)) || this;
        _this.name = "InvalidNextTokenException";
        _this.$fault = "client";
        Object.setPrototypeOf(_this, InvalidNextTokenException.prototype);
        return _this;
    }
    return InvalidNextTokenException;
}(__BaseException));
export { InvalidNextTokenException };
export var ReviewDetailsFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var AccountDetailsFilterSensitiveLog = function (obj) { return (__assign(__assign(__assign(__assign({}, obj), (obj.WebsiteURL && { WebsiteURL: SENSITIVE_STRING })), (obj.UseCaseDescription && { UseCaseDescription: SENSITIVE_STRING })), (obj.AdditionalContactEmailAddresses && { AdditionalContactEmailAddresses: SENSITIVE_STRING }))); };
export var BlacklistEntryFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ContentFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var BodyFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TemplateFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var BulkEmailContentFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DestinationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ReplacementTemplateFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ReplacementEmailContentFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var MessageTagFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var BulkEmailEntryFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var BulkEmailEntryResultFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CloudWatchDimensionConfigurationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CloudWatchDestinationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TopicPreferenceFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ContactFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ContactListFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ContactListDestinationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeliveryOptionsFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ReputationOptionsFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SendingOptionsFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SuppressionOptionsFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TagFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TrackingOptionsFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateConfigurationSetRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateConfigurationSetResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var KinesisFirehoseDestinationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PinpointDestinationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SnsDestinationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var EventDestinationDefinitionFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateConfigurationSetEventDestinationRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateConfigurationSetEventDestinationResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateContactRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateContactResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TopicFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateContactListRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateContactListResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateCustomVerificationEmailTemplateRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateCustomVerificationEmailTemplateResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateDedicatedIpPoolRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateDedicatedIpPoolResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var RawMessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var MessageFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var EmailContentFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateDeliverabilityTestReportRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateDeliverabilityTestReportResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DkimSigningAttributesFilterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.DomainSigningPrivateKey && { DomainSigningPrivateKey: SENSITIVE_STRING }))); };
export var CreateEmailIdentityRequestFilterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.DkimSigningAttributes && {
    DkimSigningAttributes: DkimSigningAttributesFilterSensitiveLog(obj.DkimSigningAttributes),
}))); };
export var DkimAttributesFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateEmailIdentityResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateEmailIdentityPolicyRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateEmailIdentityPolicyResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var EmailTemplateContentFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateEmailTemplateRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateEmailTemplateResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ImportDataSourceFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SuppressionListDestinationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ImportDestinationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateImportJobRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CreateImportJobResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var CustomVerificationEmailTemplateMetadataFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DomainIspPlacementFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var VolumeStatisticsFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DailyVolumeFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DedicatedIpFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteConfigurationSetRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteConfigurationSetResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteConfigurationSetEventDestinationRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteConfigurationSetEventDestinationResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteContactRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteContactResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteContactListRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteContactListResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteCustomVerificationEmailTemplateRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteCustomVerificationEmailTemplateResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteDedicatedIpPoolRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteDedicatedIpPoolResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteEmailIdentityRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteEmailIdentityResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteEmailIdentityPolicyRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteEmailIdentityPolicyResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteEmailTemplateRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteEmailTemplateResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteSuppressedDestinationRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeleteSuppressedDestinationResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DeliverabilityTestReportFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DomainDeliverabilityCampaignFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var InboxPlacementTrackingOptionFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var DomainDeliverabilityTrackingOptionFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var EmailTemplateMetadataFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var EventDestinationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var FailureInfoFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetAccountRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SendQuotaFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SuppressionAttributesFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetAccountResponseFilterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Details && { Details: AccountDetailsFilterSensitiveLog(obj.Details) }))); };
export var GetBlacklistReportsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetBlacklistReportsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetConfigurationSetRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetConfigurationSetResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetConfigurationSetEventDestinationsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetConfigurationSetEventDestinationsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetContactRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetContactResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetContactListRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetContactListResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetCustomVerificationEmailTemplateRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetCustomVerificationEmailTemplateResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetDedicatedIpRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetDedicatedIpResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetDedicatedIpsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetDedicatedIpsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetDeliverabilityDashboardOptionsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetDeliverabilityDashboardOptionsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetDeliverabilityTestReportRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PlacementStatisticsFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var IspPlacementFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetDeliverabilityTestReportResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetDomainDeliverabilityCampaignRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetDomainDeliverabilityCampaignResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetDomainStatisticsReportRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var OverallVolumeFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetDomainStatisticsReportResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetEmailIdentityRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var MailFromAttributesFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetEmailIdentityResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetEmailIdentityPoliciesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetEmailIdentityPoliciesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetEmailTemplateRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetEmailTemplateResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetImportJobRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetImportJobResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetSuppressedDestinationRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SuppressedDestinationAttributesFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SuppressedDestinationFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var GetSuppressedDestinationResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var IdentityInfoFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ImportJobSummaryFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListConfigurationSetsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListConfigurationSetsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListContactListsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListContactListsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TopicFilterFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListContactsFilterFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListContactsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListContactsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListCustomVerificationEmailTemplatesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListCustomVerificationEmailTemplatesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListDedicatedIpPoolsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListDedicatedIpPoolsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListDeliverabilityTestReportsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListDeliverabilityTestReportsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListDomainDeliverabilityCampaignsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListDomainDeliverabilityCampaignsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListEmailIdentitiesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListEmailIdentitiesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListEmailTemplatesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListEmailTemplatesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListImportJobsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListImportJobsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListManagementOptionsFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListSuppressedDestinationsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SuppressedDestinationSummaryFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListSuppressedDestinationsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListTagsForResourceRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var ListTagsForResourceResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutAccountDedicatedIpWarmupAttributesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutAccountDedicatedIpWarmupAttributesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutAccountDetailsRequestFilterSensitiveLog = function (obj) { return (__assign(__assign(__assign(__assign({}, obj), (obj.WebsiteURL && { WebsiteURL: SENSITIVE_STRING })), (obj.UseCaseDescription && { UseCaseDescription: SENSITIVE_STRING })), (obj.AdditionalContactEmailAddresses && { AdditionalContactEmailAddresses: SENSITIVE_STRING }))); };
export var PutAccountDetailsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutAccountSendingAttributesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutAccountSendingAttributesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutAccountSuppressionAttributesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutAccountSuppressionAttributesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutConfigurationSetDeliveryOptionsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutConfigurationSetDeliveryOptionsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutConfigurationSetReputationOptionsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutConfigurationSetReputationOptionsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutConfigurationSetSendingOptionsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutConfigurationSetSendingOptionsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutConfigurationSetSuppressionOptionsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutConfigurationSetSuppressionOptionsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutConfigurationSetTrackingOptionsRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutConfigurationSetTrackingOptionsResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutDedicatedIpInPoolRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutDedicatedIpInPoolResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutDedicatedIpWarmupAttributesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutDedicatedIpWarmupAttributesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutDeliverabilityDashboardOptionRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutDeliverabilityDashboardOptionResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutEmailIdentityConfigurationSetAttributesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutEmailIdentityConfigurationSetAttributesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutEmailIdentityDkimAttributesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutEmailIdentityDkimAttributesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutEmailIdentityDkimSigningAttributesRequestFilterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.SigningAttributes && { SigningAttributes: DkimSigningAttributesFilterSensitiveLog(obj.SigningAttributes) }))); };
export var PutEmailIdentityDkimSigningAttributesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutEmailIdentityFeedbackAttributesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutEmailIdentityFeedbackAttributesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutEmailIdentityMailFromAttributesRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutEmailIdentityMailFromAttributesResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutSuppressedDestinationRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var PutSuppressedDestinationResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SendBulkEmailRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SendBulkEmailResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SendCustomVerificationEmailRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SendCustomVerificationEmailResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SendEmailRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var SendEmailResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TagResourceRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TagResourceResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TestRenderEmailTemplateRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var TestRenderEmailTemplateResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UntagResourceRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UntagResourceResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateConfigurationSetEventDestinationRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateConfigurationSetEventDestinationResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateContactRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateContactResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateContactListRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateContactListResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateCustomVerificationEmailTemplateRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateCustomVerificationEmailTemplateResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateEmailIdentityPolicyRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateEmailIdentityPolicyResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateEmailTemplateRequestFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
export var UpdateEmailTemplateResponseFilterSensitiveLog = function (obj) { return (__assign({}, obj)); };
