// @aws-sdk/client-sesv2/dist-cjs/models/models_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudWatchDestinationFilterSensitiveLog = exports.CloudWatchDimensionConfigurationFilterSensitiveLog = exports.BulkEmailEntryResultFilterSensitiveLog = exports.BulkEmailEntryFilterSensitiveLog = exports.MessageTagFilterSensitiveLog = exports.ReplacementEmailContentFilterSensitiveLog = exports.ReplacementTemplateFilterSensitiveLog = exports.DestinationFilterSensitiveLog = exports.BulkEmailContentFilterSensitiveLog = exports.TemplateFilterSensitiveLog = exports.BodyFilterSensitiveLog = exports.ContentFilterSensitiveLog = exports.BlacklistEntryFilterSensitiveLog = exports.AccountDetailsFilterSensitiveLog = exports.ReviewDetailsFilterSensitiveLog = exports.InvalidNextTokenException = exports.ImportDestinationType = exports.JobStatus = exports.MailFromDomainStatus = exports.DeliverabilityDashboardAccountStatus = exports.WarmupStatus = exports.SuppressionListImportAction = exports.DataFormat = exports.IdentityType = exports.DkimStatus = exports.DkimSigningAttributesOrigin = exports.DkimSigningKeyLength = exports.SendingPausedException = exports.MessageRejected = exports.MailFromDomainNotVerifiedException = exports.DeliverabilityTestStatus = exports.EventType = exports.TooManyRequestsException = exports.NotFoundException = exports.LimitExceededException = exports.SuppressionListReason = exports.TlsPolicy = exports.ContactListImportAction = exports.SubscriptionStatus = exports.ConflictException = exports.ConcurrentModificationException = exports.DimensionValueSource = exports.BulkEmailStatus = exports.BehaviorOnMxFailure = exports.BadRequestException = exports.AlreadyExistsException = exports.AccountSuspendedException = exports.ReviewStatus = exports.MailType = exports.ContactLanguage = void 0;
exports.DailyVolumeFilterSensitiveLog = exports.VolumeStatisticsFilterSensitiveLog = exports.DomainIspPlacementFilterSensitiveLog = exports.CustomVerificationEmailTemplateMetadataFilterSensitiveLog = exports.CreateImportJobResponseFilterSensitiveLog = exports.CreateImportJobRequestFilterSensitiveLog = exports.ImportDestinationFilterSensitiveLog = exports.SuppressionListDestinationFilterSensitiveLog = exports.ImportDataSourceFilterSensitiveLog = exports.CreateEmailTemplateResponseFilterSensitiveLog = exports.CreateEmailTemplateRequestFilterSensitiveLog = exports.EmailTemplateContentFilterSensitiveLog = exports.CreateEmailIdentityPolicyResponseFilterSensitiveLog = exports.CreateEmailIdentityPolicyRequestFilterSensitiveLog = exports.CreateEmailIdentityResponseFilterSensitiveLog = exports.DkimAttributesFilterSensitiveLog = exports.CreateEmailIdentityRequestFilterSensitiveLog = exports.DkimSigningAttributesFilterSensitiveLog = exports.CreateDeliverabilityTestReportResponseFilterSensitiveLog = exports.CreateDeliverabilityTestReportRequestFilterSensitiveLog = exports.EmailContentFilterSensitiveLog = exports.MessageFilterSensitiveLog = exports.RawMessageFilterSensitiveLog = exports.CreateDedicatedIpPoolResponseFilterSensitiveLog = exports.CreateDedicatedIpPoolRequestFilterSensitiveLog = exports.CreateCustomVerificationEmailTemplateResponseFilterSensitiveLog = exports.CreateCustomVerificationEmailTemplateRequestFilterSensitiveLog = exports.CreateContactListResponseFilterSensitiveLog = exports.CreateContactListRequestFilterSensitiveLog = exports.TopicFilterSensitiveLog = exports.CreateContactResponseFilterSensitiveLog = exports.CreateContactRequestFilterSensitiveLog = exports.CreateConfigurationSetEventDestinationResponseFilterSensitiveLog = exports.CreateConfigurationSetEventDestinationRequestFilterSensitiveLog = exports.EventDestinationDefinitionFilterSensitiveLog = exports.SnsDestinationFilterSensitiveLog = exports.PinpointDestinationFilterSensitiveLog = exports.KinesisFirehoseDestinationFilterSensitiveLog = exports.CreateConfigurationSetResponseFilterSensitiveLog = exports.CreateConfigurationSetRequestFilterSensitiveLog = exports.TrackingOptionsFilterSensitiveLog = exports.TagFilterSensitiveLog = exports.SuppressionOptionsFilterSensitiveLog = exports.SendingOptionsFilterSensitiveLog = exports.ReputationOptionsFilterSensitiveLog = exports.DeliveryOptionsFilterSensitiveLog = exports.ContactListDestinationFilterSensitiveLog = exports.ContactListFilterSensitiveLog = exports.ContactFilterSensitiveLog = exports.TopicPreferenceFilterSensitiveLog = void 0;
exports.GetDeliverabilityDashboardOptionsResponseFilterSensitiveLog = exports.GetDeliverabilityDashboardOptionsRequestFilterSensitiveLog = exports.GetDedicatedIpsResponseFilterSensitiveLog = exports.GetDedicatedIpsRequestFilterSensitiveLog = exports.GetDedicatedIpResponseFilterSensitiveLog = exports.GetDedicatedIpRequestFilterSensitiveLog = exports.GetCustomVerificationEmailTemplateResponseFilterSensitiveLog = exports.GetCustomVerificationEmailTemplateRequestFilterSensitiveLog = exports.GetContactListResponseFilterSensitiveLog = exports.GetContactListRequestFilterSensitiveLog = exports.GetContactResponseFilterSensitiveLog = exports.GetContactRequestFilterSensitiveLog = exports.GetConfigurationSetEventDestinationsResponseFilterSensitiveLog = exports.GetConfigurationSetEventDestinationsRequestFilterSensitiveLog = exports.GetConfigurationSetResponseFilterSensitiveLog = exports.GetConfigurationSetRequestFilterSensitiveLog = exports.GetBlacklistReportsResponseFilterSensitiveLog = exports.GetBlacklistReportsRequestFilterSensitiveLog = exports.GetAccountResponseFilterSensitiveLog = exports.SuppressionAttributesFilterSensitiveLog = exports.SendQuotaFilterSensitiveLog = exports.GetAccountRequestFilterSensitiveLog = exports.FailureInfoFilterSensitiveLog = exports.EventDestinationFilterSensitiveLog = exports.EmailTemplateMetadataFilterSensitiveLog = exports.DomainDeliverabilityTrackingOptionFilterSensitiveLog = exports.InboxPlacementTrackingOptionFilterSensitiveLog = exports.DomainDeliverabilityCampaignFilterSensitiveLog = exports.DeliverabilityTestReportFilterSensitiveLog = exports.DeleteSuppressedDestinationResponseFilterSensitiveLog = exports.DeleteSuppressedDestinationRequestFilterSensitiveLog = exports.DeleteEmailTemplateResponseFilterSensitiveLog = exports.DeleteEmailTemplateRequestFilterSensitiveLog = exports.DeleteEmailIdentityPolicyResponseFilterSensitiveLog = exports.DeleteEmailIdentityPolicyRequestFilterSensitiveLog = exports.DeleteEmailIdentityResponseFilterSensitiveLog = exports.DeleteEmailIdentityRequestFilterSensitiveLog = exports.DeleteDedicatedIpPoolResponseFilterSensitiveLog = exports.DeleteDedicatedIpPoolRequestFilterSensitiveLog = exports.DeleteCustomVerificationEmailTemplateResponseFilterSensitiveLog = exports.DeleteCustomVerificationEmailTemplateRequestFilterSensitiveLog = exports.DeleteContactListResponseFilterSensitiveLog = exports.DeleteContactListRequestFilterSensitiveLog = exports.DeleteContactResponseFilterSensitiveLog = exports.DeleteContactRequestFilterSensitiveLog = exports.DeleteConfigurationSetEventDestinationResponseFilterSensitiveLog = exports.DeleteConfigurationSetEventDestinationRequestFilterSensitiveLog = exports.DeleteConfigurationSetResponseFilterSensitiveLog = exports.DeleteConfigurationSetRequestFilterSensitiveLog = exports.DedicatedIpFilterSensitiveLog = void 0;
exports.ListSuppressedDestinationsResponseFilterSensitiveLog = exports.SuppressedDestinationSummaryFilterSensitiveLog = exports.ListSuppressedDestinationsRequestFilterSensitiveLog = exports.ListManagementOptionsFilterSensitiveLog = exports.ListImportJobsResponseFilterSensitiveLog = exports.ListImportJobsRequestFilterSensitiveLog = exports.ListEmailTemplatesResponseFilterSensitiveLog = exports.ListEmailTemplatesRequestFilterSensitiveLog = exports.ListEmailIdentitiesResponseFilterSensitiveLog = exports.ListEmailIdentitiesRequestFilterSensitiveLog = exports.ListDomainDeliverabilityCampaignsResponseFilterSensitiveLog = exports.ListDomainDeliverabilityCampaignsRequestFilterSensitiveLog = exports.ListDeliverabilityTestReportsResponseFilterSensitiveLog = exports.ListDeliverabilityTestReportsRequestFilterSensitiveLog = exports.ListDedicatedIpPoolsResponseFilterSensitiveLog = exports.ListDedicatedIpPoolsRequestFilterSensitiveLog = exports.ListCustomVerificationEmailTemplatesResponseFilterSensitiveLog = exports.ListCustomVerificationEmailTemplatesRequestFilterSensitiveLog = exports.ListContactsResponseFilterSensitiveLog = exports.ListContactsRequestFilterSensitiveLog = exports.ListContactsFilterFilterSensitiveLog = exports.TopicFilterFilterSensitiveLog = exports.ListContactListsResponseFilterSensitiveLog = exports.ListContactListsRequestFilterSensitiveLog = exports.ListConfigurationSetsResponseFilterSensitiveLog = exports.ListConfigurationSetsRequestFilterSensitiveLog = exports.ImportJobSummaryFilterSensitiveLog = exports.IdentityInfoFilterSensitiveLog = exports.GetSuppressedDestinationResponseFilterSensitiveLog = exports.SuppressedDestinationFilterSensitiveLog = exports.SuppressedDestinationAttributesFilterSensitiveLog = exports.GetSuppressedDestinationRequestFilterSensitiveLog = exports.GetImportJobResponseFilterSensitiveLog = exports.GetImportJobRequestFilterSensitiveLog = exports.GetEmailTemplateResponseFilterSensitiveLog = exports.GetEmailTemplateRequestFilterSensitiveLog = exports.GetEmailIdentityPoliciesResponseFilterSensitiveLog = exports.GetEmailIdentityPoliciesRequestFilterSensitiveLog = exports.GetEmailIdentityResponseFilterSensitiveLog = exports.MailFromAttributesFilterSensitiveLog = exports.GetEmailIdentityRequestFilterSensitiveLog = exports.GetDomainStatisticsReportResponseFilterSensitiveLog = exports.OverallVolumeFilterSensitiveLog = exports.GetDomainStatisticsReportRequestFilterSensitiveLog = exports.GetDomainDeliverabilityCampaignResponseFilterSensitiveLog = exports.GetDomainDeliverabilityCampaignRequestFilterSensitiveLog = exports.GetDeliverabilityTestReportResponseFilterSensitiveLog = exports.IspPlacementFilterSensitiveLog = exports.PlacementStatisticsFilterSensitiveLog = exports.GetDeliverabilityTestReportRequestFilterSensitiveLog = void 0;
exports.UntagResourceResponseFilterSensitiveLog = exports.UntagResourceRequestFilterSensitiveLog = exports.TestRenderEmailTemplateResponseFilterSensitiveLog = exports.TestRenderEmailTemplateRequestFilterSensitiveLog = exports.TagResourceResponseFilterSensitiveLog = exports.TagResourceRequestFilterSensitiveLog = exports.SendEmailResponseFilterSensitiveLog = exports.SendEmailRequestFilterSensitiveLog = exports.SendCustomVerificationEmailResponseFilterSensitiveLog = exports.SendCustomVerificationEmailRequestFilterSensitiveLog = exports.SendBulkEmailResponseFilterSensitiveLog = exports.SendBulkEmailRequestFilterSensitiveLog = exports.PutSuppressedDestinationResponseFilterSensitiveLog = exports.PutSuppressedDestinationRequestFilterSensitiveLog = exports.PutEmailIdentityMailFromAttributesResponseFilterSensitiveLog = exports.PutEmailIdentityMailFromAttributesRequestFilterSensitiveLog = exports.PutEmailIdentityFeedbackAttributesResponseFilterSensitiveLog = exports.PutEmailIdentityFeedbackAttributesRequestFilterSensitiveLog = exports.PutEmailIdentityDkimSigningAttributesResponseFilterSensitiveLog = exports.PutEmailIdentityDkimSigningAttributesRequestFilterSensitiveLog = exports.PutEmailIdentityDkimAttributesResponseFilterSensitiveLog = exports.PutEmailIdentityDkimAttributesRequestFilterSensitiveLog = exports.PutEmailIdentityConfigurationSetAttributesResponseFilterSensitiveLog = exports.PutEmailIdentityConfigurationSetAttributesRequestFilterSensitiveLog = exports.PutDeliverabilityDashboardOptionResponseFilterSensitiveLog = exports.PutDeliverabilityDashboardOptionRequestFilterSensitiveLog = exports.PutDedicatedIpWarmupAttributesResponseFilterSensitiveLog = exports.PutDedicatedIpWarmupAttributesRequestFilterSensitiveLog = exports.PutDedicatedIpInPoolResponseFilterSensitiveLog = exports.PutDedicatedIpInPoolRequestFilterSensitiveLog = exports.PutConfigurationSetTrackingOptionsResponseFilterSensitiveLog = exports.PutConfigurationSetTrackingOptionsRequestFilterSensitiveLog = exports.PutConfigurationSetSuppressionOptionsResponseFilterSensitiveLog = exports.PutConfigurationSetSuppressionOptionsRequestFilterSensitiveLog = exports.PutConfigurationSetSendingOptionsResponseFilterSensitiveLog = exports.PutConfigurationSetSendingOptionsRequestFilterSensitiveLog = exports.PutConfigurationSetReputationOptionsResponseFilterSensitiveLog = exports.PutConfigurationSetReputationOptionsRequestFilterSensitiveLog = exports.PutConfigurationSetDeliveryOptionsResponseFilterSensitiveLog = exports.PutConfigurationSetDeliveryOptionsRequestFilterSensitiveLog = exports.PutAccountSuppressionAttributesResponseFilterSensitiveLog = exports.PutAccountSuppressionAttributesRequestFilterSensitiveLog = exports.PutAccountSendingAttributesResponseFilterSensitiveLog = exports.PutAccountSendingAttributesRequestFilterSensitiveLog = exports.PutAccountDetailsResponseFilterSensitiveLog = exports.PutAccountDetailsRequestFilterSensitiveLog = exports.PutAccountDedicatedIpWarmupAttributesResponseFilterSensitiveLog = exports.PutAccountDedicatedIpWarmupAttributesRequestFilterSensitiveLog = exports.ListTagsForResourceResponseFilterSensitiveLog = exports.ListTagsForResourceRequestFilterSensitiveLog = void 0;
exports.UpdateEmailTemplateResponseFilterSensitiveLog = exports.UpdateEmailTemplateRequestFilterSensitiveLog = exports.UpdateEmailIdentityPolicyResponseFilterSensitiveLog = exports.UpdateEmailIdentityPolicyRequestFilterSensitiveLog = exports.UpdateCustomVerificationEmailTemplateResponseFilterSensitiveLog = exports.UpdateCustomVerificationEmailTemplateRequestFilterSensitiveLog = exports.UpdateContactListResponseFilterSensitiveLog = exports.UpdateContactListRequestFilterSensitiveLog = exports.UpdateContactResponseFilterSensitiveLog = exports.UpdateContactRequestFilterSensitiveLog = exports.UpdateConfigurationSetEventDestinationResponseFilterSensitiveLog = exports.UpdateConfigurationSetEventDestinationRequestFilterSensitiveLog = void 0;
const smithy_client_1 = require("@aws-sdk/smithy-client");
const SESv2ServiceException_1 = require("./SESv2ServiceException");
var ContactLanguage;
(function (ContactLanguage) {
    ContactLanguage["EN"] = "EN";
    ContactLanguage["JA"] = "JA";
})(ContactLanguage = exports.ContactLanguage || (exports.ContactLanguage = {}));
var MailType;
(function (MailType) {
    MailType["MARKETING"] = "MARKETING";
    MailType["TRANSACTIONAL"] = "TRANSACTIONAL";
})(MailType = exports.MailType || (exports.MailType = {}));
var ReviewStatus;
(function (ReviewStatus) {
    ReviewStatus["DENIED"] = "DENIED";
    ReviewStatus["FAILED"] = "FAILED";
    ReviewStatus["GRANTED"] = "GRANTED";
    ReviewStatus["PENDING"] = "PENDING";
})(ReviewStatus = exports.ReviewStatus || (exports.ReviewStatus = {}));
class AccountSuspendedException extends SESv2ServiceException_1.SESv2ServiceException {
    constructor(opts) {
        super({
            name: "AccountSuspendedException",
            $fault: "client",
            ...opts,
        });
        this.name = "AccountSuspendedException";
        this.$fault = "client";
        Object.setPrototypeOf(this, AccountSuspendedException.prototype);
    }
}
exports.AccountSuspendedException = AccountSuspendedException;
class AlreadyExistsException extends SESv2ServiceException_1.SESv2ServiceException {
    constructor(opts) {
        super({
            name: "AlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        this.name = "AlreadyExistsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, AlreadyExistsException.prototype);
    }
}
exports.AlreadyExistsException = AlreadyExistsException;
class BadRequestException extends SESv2ServiceException_1.SESv2ServiceException {
    constructor(opts) {
        super({
            name: "BadRequestException",
            $fault: "client",
            ...opts,
        });
        this.name = "BadRequestException";
        this.$fault = "client";
        Object.setPrototypeOf(this, BadRequestException.prototype);
    }
}
exports.BadRequestException = BadRequestException;
var BehaviorOnMxFailure;
(function (BehaviorOnMxFailure) {
    BehaviorOnMxFailure["REJECT_MESSAGE"] = "REJECT_MESSAGE";
    BehaviorOnMxFailure["USE_DEFAULT_VALUE"] = "USE_DEFAULT_VALUE";
})(BehaviorOnMxFailure = exports.BehaviorOnMxFailure || (exports.BehaviorOnMxFailure = {}));
var BulkEmailStatus;
(function (BulkEmailStatus) {
    BulkEmailStatus["ACCOUNT_DAILY_QUOTA_EXCEEDED"] = "ACCOUNT_DAILY_QUOTA_EXCEEDED";
    BulkEmailStatus["ACCOUNT_SENDING_PAUSED"] = "ACCOUNT_SENDING_PAUSED";
    BulkEmailStatus["ACCOUNT_SUSPENDED"] = "ACCOUNT_SUSPENDED";
    BulkEmailStatus["ACCOUNT_THROTTLED"] = "ACCOUNT_THROTTLED";
    BulkEmailStatus["CONFIGURATION_SET_NOT_FOUND"] = "CONFIGURATION_SET_NOT_FOUND";
    BulkEmailStatus["CONFIGURATION_SET_SENDING_PAUSED"] = "CONFIGURATION_SET_SENDING_PAUSED";
    BulkEmailStatus["FAILED"] = "FAILED";
    BulkEmailStatus["INVALID_PARAMETER"] = "INVALID_PARAMETER";
    BulkEmailStatus["INVALID_SENDING_POOL_NAME"] = "INVALID_SENDING_POOL_NAME";
    BulkEmailStatus["MAIL_FROM_DOMAIN_NOT_VERIFIED"] = "MAIL_FROM_DOMAIN_NOT_VERIFIED";
    BulkEmailStatus["MESSAGE_REJECTED"] = "MESSAGE_REJECTED";
    BulkEmailStatus["SUCCESS"] = "SUCCESS";
    BulkEmailStatus["TEMPLATE_NOT_FOUND"] = "TEMPLATE_NOT_FOUND";
    BulkEmailStatus["TRANSIENT_FAILURE"] = "TRANSIENT_FAILURE";
})(BulkEmailStatus = exports.BulkEmailStatus || (exports.BulkEmailStatus = {}));
var DimensionValueSource;
(function (DimensionValueSource) {
    DimensionValueSource["EMAIL_HEADER"] = "EMAIL_HEADER";
    DimensionValueSource["LINK_TAG"] = "LINK_TAG";
    DimensionValueSource["MESSAGE_TAG"] = "MESSAGE_TAG";
})(DimensionValueSource = exports.DimensionValueSource || (exports.DimensionValueSource = {}));
class ConcurrentModificationException extends SESv2ServiceException_1.SESv2ServiceException {
    constructor(opts) {
        super({
            name: "ConcurrentModificationException",
            $fault: "server",
            ...opts,
        });
        this.name = "ConcurrentModificationException";
        this.$fault = "server";
        Object.setPrototypeOf(this, ConcurrentModificationException.prototype);
    }
}
exports.ConcurrentModificationException = ConcurrentModificationException;
class ConflictException extends SESv2ServiceException_1.SESv2ServiceException {
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        this.name = "ConflictException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ConflictException.prototype);
    }
}
exports.ConflictException = ConflictException;
var SubscriptionStatus;
(function (SubscriptionStatus) {
    SubscriptionStatus["OPT_IN"] = "OPT_IN";
    SubscriptionStatus["OPT_OUT"] = "OPT_OUT";
})(SubscriptionStatus = exports.SubscriptionStatus || (exports.SubscriptionStatus = {}));
var ContactListImportAction;
(function (ContactListImportAction) {
    ContactListImportAction["DELETE"] = "DELETE";
    ContactListImportAction["PUT"] = "PUT";
})(ContactListImportAction = exports.ContactListImportAction || (exports.ContactListImportAction = {}));
var TlsPolicy;
(function (TlsPolicy) {
    TlsPolicy["OPTIONAL"] = "OPTIONAL";
    TlsPolicy["REQUIRE"] = "REQUIRE";
})(TlsPolicy = exports.TlsPolicy || (exports.TlsPolicy = {}));
var SuppressionListReason;
(function (SuppressionListReason) {
    SuppressionListReason["BOUNCE"] = "BOUNCE";
    SuppressionListReason["COMPLAINT"] = "COMPLAINT";
})(SuppressionListReason = exports.SuppressionListReason || (exports.SuppressionListReason = {}));
class LimitExceededException extends SESv2ServiceException_1.SESv2ServiceException {
    constructor(opts) {
        super({
            name: "LimitExceededException",
            $fault: "client",
            ...opts,
        });
        this.name = "LimitExceededException";
        this.$fault = "client";
        Object.setPrototypeOf(this, LimitExceededException.prototype);
    }
}
exports.LimitExceededException = LimitExceededException;
class NotFoundException extends SESv2ServiceException_1.SESv2ServiceException {
    constructor(opts) {
        super({
            name: "NotFoundException",
            $fault: "client",
            ...opts,
        });
        this.name = "NotFoundException";
        this.$fault = "client";
        Object.setPrototypeOf(this, NotFoundException.prototype);
    }
}
exports.NotFoundException = NotFoundException;
class TooManyRequestsException extends SESv2ServiceException_1.SESv2ServiceException {
    constructor(opts) {
        super({
            name: "TooManyRequestsException",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyRequestsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyRequestsException.prototype);
    }
}
exports.TooManyRequestsException = TooManyRequestsException;
var EventType;
(function (EventType) {
    EventType["BOUNCE"] = "BOUNCE";
    EventType["CLICK"] = "CLICK";
    EventType["COMPLAINT"] = "COMPLAINT";
    EventType["DELIVERY"] = "DELIVERY";
    EventType["DELIVERY_DELAY"] = "DELIVERY_DELAY";
    EventType["OPEN"] = "OPEN";
    EventType["REJECT"] = "REJECT";
    EventType["RENDERING_FAILURE"] = "RENDERING_FAILURE";
    EventType["SEND"] = "SEND";
    EventType["SUBSCRIPTION"] = "SUBSCRIPTION";
})(EventType = exports.EventType || (exports.EventType = {}));
var DeliverabilityTestStatus;
(function (DeliverabilityTestStatus) {
    DeliverabilityTestStatus["COMPLETED"] = "COMPLETED";
    DeliverabilityTestStatus["IN_PROGRESS"] = "IN_PROGRESS";
})(DeliverabilityTestStatus = exports.DeliverabilityTestStatus || (exports.DeliverabilityTestStatus = {}));
class MailFromDomainNotVerifiedException extends SESv2ServiceException_1.SESv2ServiceException {
    constructor(opts) {
        super({
            name: "MailFromDomainNotVerifiedException",
            $fault: "client",
            ...opts,
        });
        this.name = "MailFromDomainNotVerifiedException";
        this.$fault = "client";
        Object.setPrototypeOf(this, MailFromDomainNotVerifiedException.prototype);
    }
}
exports.MailFromDomainNotVerifiedException = MailFromDomainNotVerifiedException;
class MessageRejected extends SESv2ServiceException_1.SESv2ServiceException {
    constructor(opts) {
        super({
            name: "MessageRejected",
            $fault: "client",
            ...opts,
        });
        this.name = "MessageRejected";
        this.$fault = "client";
        Object.setPrototypeOf(this, MessageRejected.prototype);
    }
}
exports.MessageRejected = MessageRejected;
class SendingPausedException extends SESv2ServiceException_1.SESv2ServiceException {
    constructor(opts) {
        super({
            name: "SendingPausedException",
            $fault: "client",
            ...opts,
        });
        this.name = "SendingPausedException";
        this.$fault = "client";
        Object.setPrototypeOf(this, SendingPausedException.prototype);
    }
}
exports.SendingPausedException = SendingPausedException;
var DkimSigningKeyLength;
(function (DkimSigningKeyLength) {
    DkimSigningKeyLength["RSA_1024_BIT"] = "RSA_1024_BIT";
    DkimSigningKeyLength["RSA_2048_BIT"] = "RSA_2048_BIT";
})(DkimSigningKeyLength = exports.DkimSigningKeyLength || (exports.DkimSigningKeyLength = {}));
var DkimSigningAttributesOrigin;
(function (DkimSigningAttributesOrigin) {
    DkimSigningAttributesOrigin["AWS_SES"] = "AWS_SES";
    DkimSigningAttributesOrigin["EXTERNAL"] = "EXTERNAL";
})(DkimSigningAttributesOrigin = exports.DkimSigningAttributesOrigin || (exports.DkimSigningAttributesOrigin = {}));
var DkimStatus;
(function (DkimStatus) {
    DkimStatus["FAILED"] = "FAILED";
    DkimStatus["NOT_STARTED"] = "NOT_STARTED";
    DkimStatus["PENDING"] = "PENDING";
    DkimStatus["SUCCESS"] = "SUCCESS";
    DkimStatus["TEMPORARY_FAILURE"] = "TEMPORARY_FAILURE";
})(DkimStatus = exports.DkimStatus || (exports.DkimStatus = {}));
var IdentityType;
(function (IdentityType) {
    IdentityType["DOMAIN"] = "DOMAIN";
    IdentityType["EMAIL_ADDRESS"] = "EMAIL_ADDRESS";
    IdentityType["MANAGED_DOMAIN"] = "MANAGED_DOMAIN";
})(IdentityType = exports.IdentityType || (exports.IdentityType = {}));
var DataFormat;
(function (DataFormat) {
    DataFormat["CSV"] = "CSV";
    DataFormat["JSON"] = "JSON";
})(DataFormat = exports.DataFormat || (exports.DataFormat = {}));
var SuppressionListImportAction;
(function (SuppressionListImportAction) {
    SuppressionListImportAction["DELETE"] = "DELETE";
    SuppressionListImportAction["PUT"] = "PUT";
})(SuppressionListImportAction = exports.SuppressionListImportAction || (exports.SuppressionListImportAction = {}));
var WarmupStatus;
(function (WarmupStatus) {
    WarmupStatus["DONE"] = "DONE";
    WarmupStatus["IN_PROGRESS"] = "IN_PROGRESS";
})(WarmupStatus = exports.WarmupStatus || (exports.WarmupStatus = {}));
var DeliverabilityDashboardAccountStatus;
(function (DeliverabilityDashboardAccountStatus) {
    DeliverabilityDashboardAccountStatus["ACTIVE"] = "ACTIVE";
    DeliverabilityDashboardAccountStatus["DISABLED"] = "DISABLED";
    DeliverabilityDashboardAccountStatus["PENDING_EXPIRATION"] = "PENDING_EXPIRATION";
})(DeliverabilityDashboardAccountStatus = exports.DeliverabilityDashboardAccountStatus || (exports.DeliverabilityDashboardAccountStatus = {}));
var MailFromDomainStatus;
(function (MailFromDomainStatus) {
    MailFromDomainStatus["FAILED"] = "FAILED";
    MailFromDomainStatus["PENDING"] = "PENDING";
    MailFromDomainStatus["SUCCESS"] = "SUCCESS";
    MailFromDomainStatus["TEMPORARY_FAILURE"] = "TEMPORARY_FAILURE";
})(MailFromDomainStatus = exports.MailFromDomainStatus || (exports.MailFromDomainStatus = {}));
var JobStatus;
(function (JobStatus) {
    JobStatus["COMPLETED"] = "COMPLETED";
    JobStatus["CREATED"] = "CREATED";
    JobStatus["FAILED"] = "FAILED";
    JobStatus["PROCESSING"] = "PROCESSING";
})(JobStatus = exports.JobStatus || (exports.JobStatus = {}));
var ImportDestinationType;
(function (ImportDestinationType) {
    ImportDestinationType["CONTACT_LIST"] = "CONTACT_LIST";
    ImportDestinationType["SUPPRESSION_LIST"] = "SUPPRESSION_LIST";
})(ImportDestinationType = exports.ImportDestinationType || (exports.ImportDestinationType = {}));
class InvalidNextTokenException extends SESv2ServiceException_1.SESv2ServiceException {
    constructor(opts) {
        super({
            name: "InvalidNextTokenException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidNextTokenException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidNextTokenException.prototype);
    }
}
exports.InvalidNextTokenException = InvalidNextTokenException;
const ReviewDetailsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ReviewDetailsFilterSensitiveLog = ReviewDetailsFilterSensitiveLog;
const AccountDetailsFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.WebsiteURL && { WebsiteURL: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.UseCaseDescription && { UseCaseDescription: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.AdditionalContactEmailAddresses && { AdditionalContactEmailAddresses: smithy_client_1.SENSITIVE_STRING }),
});
exports.AccountDetailsFilterSensitiveLog = AccountDetailsFilterSensitiveLog;
const BlacklistEntryFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BlacklistEntryFilterSensitiveLog = BlacklistEntryFilterSensitiveLog;
const ContentFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ContentFilterSensitiveLog = ContentFilterSensitiveLog;
const BodyFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BodyFilterSensitiveLog = BodyFilterSensitiveLog;
const TemplateFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TemplateFilterSensitiveLog = TemplateFilterSensitiveLog;
const BulkEmailContentFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BulkEmailContentFilterSensitiveLog = BulkEmailContentFilterSensitiveLog;
const DestinationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DestinationFilterSensitiveLog = DestinationFilterSensitiveLog;
const ReplacementTemplateFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ReplacementTemplateFilterSensitiveLog = ReplacementTemplateFilterSensitiveLog;
const ReplacementEmailContentFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ReplacementEmailContentFilterSensitiveLog = ReplacementEmailContentFilterSensitiveLog;
const MessageTagFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.MessageTagFilterSensitiveLog = MessageTagFilterSensitiveLog;
const BulkEmailEntryFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BulkEmailEntryFilterSensitiveLog = BulkEmailEntryFilterSensitiveLog;
const BulkEmailEntryResultFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BulkEmailEntryResultFilterSensitiveLog = BulkEmailEntryResultFilterSensitiveLog;
const CloudWatchDimensionConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CloudWatchDimensionConfigurationFilterSensitiveLog = CloudWatchDimensionConfigurationFilterSensitiveLog;
const CloudWatchDestinationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CloudWatchDestinationFilterSensitiveLog = CloudWatchDestinationFilterSensitiveLog;
const TopicPreferenceFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TopicPreferenceFilterSensitiveLog = TopicPreferenceFilterSensitiveLog;
const ContactFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ContactFilterSensitiveLog = ContactFilterSensitiveLog;
const ContactListFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ContactListFilterSensitiveLog = ContactListFilterSensitiveLog;
const ContactListDestinationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ContactListDestinationFilterSensitiveLog = ContactListDestinationFilterSensitiveLog;
const DeliveryOptionsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeliveryOptionsFilterSensitiveLog = DeliveryOptionsFilterSensitiveLog;
const ReputationOptionsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ReputationOptionsFilterSensitiveLog = ReputationOptionsFilterSensitiveLog;
const SendingOptionsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SendingOptionsFilterSensitiveLog = SendingOptionsFilterSensitiveLog;
const SuppressionOptionsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SuppressionOptionsFilterSensitiveLog = SuppressionOptionsFilterSensitiveLog;
const TagFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TagFilterSensitiveLog = TagFilterSensitiveLog;
const TrackingOptionsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TrackingOptionsFilterSensitiveLog = TrackingOptionsFilterSensitiveLog;
const CreateConfigurationSetRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateConfigurationSetRequestFilterSensitiveLog = CreateConfigurationSetRequestFilterSensitiveLog;
const CreateConfigurationSetResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateConfigurationSetResponseFilterSensitiveLog = CreateConfigurationSetResponseFilterSensitiveLog;
const KinesisFirehoseDestinationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.KinesisFirehoseDestinationFilterSensitiveLog = KinesisFirehoseDestinationFilterSensitiveLog;
const PinpointDestinationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PinpointDestinationFilterSensitiveLog = PinpointDestinationFilterSensitiveLog;
const SnsDestinationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SnsDestinationFilterSensitiveLog = SnsDestinationFilterSensitiveLog;
const EventDestinationDefinitionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EventDestinationDefinitionFilterSensitiveLog = EventDestinationDefinitionFilterSensitiveLog;
const CreateConfigurationSetEventDestinationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateConfigurationSetEventDestinationRequestFilterSensitiveLog = CreateConfigurationSetEventDestinationRequestFilterSensitiveLog;
const CreateConfigurationSetEventDestinationResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateConfigurationSetEventDestinationResponseFilterSensitiveLog = CreateConfigurationSetEventDestinationResponseFilterSensitiveLog;
const CreateContactRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateContactRequestFilterSensitiveLog = CreateContactRequestFilterSensitiveLog;
const CreateContactResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateContactResponseFilterSensitiveLog = CreateContactResponseFilterSensitiveLog;
const TopicFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TopicFilterSensitiveLog = TopicFilterSensitiveLog;
const CreateContactListRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateContactListRequestFilterSensitiveLog = CreateContactListRequestFilterSensitiveLog;
const CreateContactListResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateContactListResponseFilterSensitiveLog = CreateContactListResponseFilterSensitiveLog;
const CreateCustomVerificationEmailTemplateRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateCustomVerificationEmailTemplateRequestFilterSensitiveLog = CreateCustomVerificationEmailTemplateRequestFilterSensitiveLog;
const CreateCustomVerificationEmailTemplateResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateCustomVerificationEmailTemplateResponseFilterSensitiveLog = CreateCustomVerificationEmailTemplateResponseFilterSensitiveLog;
const CreateDedicatedIpPoolRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateDedicatedIpPoolRequestFilterSensitiveLog = CreateDedicatedIpPoolRequestFilterSensitiveLog;
const CreateDedicatedIpPoolResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateDedicatedIpPoolResponseFilterSensitiveLog = CreateDedicatedIpPoolResponseFilterSensitiveLog;
const RawMessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.RawMessageFilterSensitiveLog = RawMessageFilterSensitiveLog;
const MessageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.MessageFilterSensitiveLog = MessageFilterSensitiveLog;
const EmailContentFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EmailContentFilterSensitiveLog = EmailContentFilterSensitiveLog;
const CreateDeliverabilityTestReportRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateDeliverabilityTestReportRequestFilterSensitiveLog = CreateDeliverabilityTestReportRequestFilterSensitiveLog;
const CreateDeliverabilityTestReportResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateDeliverabilityTestReportResponseFilterSensitiveLog = CreateDeliverabilityTestReportResponseFilterSensitiveLog;
const DkimSigningAttributesFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.DomainSigningPrivateKey && { DomainSigningPrivateKey: smithy_client_1.SENSITIVE_STRING }),
});
exports.DkimSigningAttributesFilterSensitiveLog = DkimSigningAttributesFilterSensitiveLog;
const CreateEmailIdentityRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.DkimSigningAttributes && {
        DkimSigningAttributes: (0, exports.DkimSigningAttributesFilterSensitiveLog)(obj.DkimSigningAttributes),
    }),
});
exports.CreateEmailIdentityRequestFilterSensitiveLog = CreateEmailIdentityRequestFilterSensitiveLog;
const DkimAttributesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DkimAttributesFilterSensitiveLog = DkimAttributesFilterSensitiveLog;
const CreateEmailIdentityResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateEmailIdentityResponseFilterSensitiveLog = CreateEmailIdentityResponseFilterSensitiveLog;
const CreateEmailIdentityPolicyRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateEmailIdentityPolicyRequestFilterSensitiveLog = CreateEmailIdentityPolicyRequestFilterSensitiveLog;
const CreateEmailIdentityPolicyResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateEmailIdentityPolicyResponseFilterSensitiveLog = CreateEmailIdentityPolicyResponseFilterSensitiveLog;
const EmailTemplateContentFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EmailTemplateContentFilterSensitiveLog = EmailTemplateContentFilterSensitiveLog;
const CreateEmailTemplateRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateEmailTemplateRequestFilterSensitiveLog = CreateEmailTemplateRequestFilterSensitiveLog;
const CreateEmailTemplateResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateEmailTemplateResponseFilterSensitiveLog = CreateEmailTemplateResponseFilterSensitiveLog;
const ImportDataSourceFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ImportDataSourceFilterSensitiveLog = ImportDataSourceFilterSensitiveLog;
const SuppressionListDestinationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SuppressionListDestinationFilterSensitiveLog = SuppressionListDestinationFilterSensitiveLog;
const ImportDestinationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ImportDestinationFilterSensitiveLog = ImportDestinationFilterSensitiveLog;
const CreateImportJobRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateImportJobRequestFilterSensitiveLog = CreateImportJobRequestFilterSensitiveLog;
const CreateImportJobResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateImportJobResponseFilterSensitiveLog = CreateImportJobResponseFilterSensitiveLog;
const CustomVerificationEmailTemplateMetadataFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CustomVerificationEmailTemplateMetadataFilterSensitiveLog = CustomVerificationEmailTemplateMetadataFilterSensitiveLog;
const DomainIspPlacementFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DomainIspPlacementFilterSensitiveLog = DomainIspPlacementFilterSensitiveLog;
const VolumeStatisticsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.VolumeStatisticsFilterSensitiveLog = VolumeStatisticsFilterSensitiveLog;
const DailyVolumeFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DailyVolumeFilterSensitiveLog = DailyVolumeFilterSensitiveLog;
const DedicatedIpFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DedicatedIpFilterSensitiveLog = DedicatedIpFilterSensitiveLog;
const DeleteConfigurationSetRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteConfigurationSetRequestFilterSensitiveLog = DeleteConfigurationSetRequestFilterSensitiveLog;
const DeleteConfigurationSetResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteConfigurationSetResponseFilterSensitiveLog = DeleteConfigurationSetResponseFilterSensitiveLog;
const DeleteConfigurationSetEventDestinationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteConfigurationSetEventDestinationRequestFilterSensitiveLog = DeleteConfigurationSetEventDestinationRequestFilterSensitiveLog;
const DeleteConfigurationSetEventDestinationResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteConfigurationSetEventDestinationResponseFilterSensitiveLog = DeleteConfigurationSetEventDestinationResponseFilterSensitiveLog;
const DeleteContactRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteContactRequestFilterSensitiveLog = DeleteContactRequestFilterSensitiveLog;
const DeleteContactResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteContactResponseFilterSensitiveLog = DeleteContactResponseFilterSensitiveLog;
const DeleteContactListRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteContactListRequestFilterSensitiveLog = DeleteContactListRequestFilterSensitiveLog;
const DeleteContactListResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteContactListResponseFilterSensitiveLog = DeleteContactListResponseFilterSensitiveLog;
const DeleteCustomVerificationEmailTemplateRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteCustomVerificationEmailTemplateRequestFilterSensitiveLog = DeleteCustomVerificationEmailTemplateRequestFilterSensitiveLog;
const DeleteCustomVerificationEmailTemplateResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteCustomVerificationEmailTemplateResponseFilterSensitiveLog = DeleteCustomVerificationEmailTemplateResponseFilterSensitiveLog;
const DeleteDedicatedIpPoolRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteDedicatedIpPoolRequestFilterSensitiveLog = DeleteDedicatedIpPoolRequestFilterSensitiveLog;
const DeleteDedicatedIpPoolResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteDedicatedIpPoolResponseFilterSensitiveLog = DeleteDedicatedIpPoolResponseFilterSensitiveLog;
const DeleteEmailIdentityRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteEmailIdentityRequestFilterSensitiveLog = DeleteEmailIdentityRequestFilterSensitiveLog;
const DeleteEmailIdentityResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteEmailIdentityResponseFilterSensitiveLog = DeleteEmailIdentityResponseFilterSensitiveLog;
const DeleteEmailIdentityPolicyRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteEmailIdentityPolicyRequestFilterSensitiveLog = DeleteEmailIdentityPolicyRequestFilterSensitiveLog;
const DeleteEmailIdentityPolicyResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteEmailIdentityPolicyResponseFilterSensitiveLog = DeleteEmailIdentityPolicyResponseFilterSensitiveLog;
const DeleteEmailTemplateRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteEmailTemplateRequestFilterSensitiveLog = DeleteEmailTemplateRequestFilterSensitiveLog;
const DeleteEmailTemplateResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteEmailTemplateResponseFilterSensitiveLog = DeleteEmailTemplateResponseFilterSensitiveLog;
const DeleteSuppressedDestinationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteSuppressedDestinationRequestFilterSensitiveLog = DeleteSuppressedDestinationRequestFilterSensitiveLog;
const DeleteSuppressedDestinationResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteSuppressedDestinationResponseFilterSensitiveLog = DeleteSuppressedDestinationResponseFilterSensitiveLog;
const DeliverabilityTestReportFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeliverabilityTestReportFilterSensitiveLog = DeliverabilityTestReportFilterSensitiveLog;
const DomainDeliverabilityCampaignFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DomainDeliverabilityCampaignFilterSensitiveLog = DomainDeliverabilityCampaignFilterSensitiveLog;
const InboxPlacementTrackingOptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.InboxPlacementTrackingOptionFilterSensitiveLog = InboxPlacementTrackingOptionFilterSensitiveLog;
const DomainDeliverabilityTrackingOptionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DomainDeliverabilityTrackingOptionFilterSensitiveLog = DomainDeliverabilityTrackingOptionFilterSensitiveLog;
const EmailTemplateMetadataFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EmailTemplateMetadataFilterSensitiveLog = EmailTemplateMetadataFilterSensitiveLog;
const EventDestinationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EventDestinationFilterSensitiveLog = EventDestinationFilterSensitiveLog;
const FailureInfoFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.FailureInfoFilterSensitiveLog = FailureInfoFilterSensitiveLog;
const GetAccountRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetAccountRequestFilterSensitiveLog = GetAccountRequestFilterSensitiveLog;
const SendQuotaFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SendQuotaFilterSensitiveLog = SendQuotaFilterSensitiveLog;
const SuppressionAttributesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SuppressionAttributesFilterSensitiveLog = SuppressionAttributesFilterSensitiveLog;
const GetAccountResponseFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.Details && { Details: (0, exports.AccountDetailsFilterSensitiveLog)(obj.Details) }),
});
exports.GetAccountResponseFilterSensitiveLog = GetAccountResponseFilterSensitiveLog;
const GetBlacklistReportsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetBlacklistReportsRequestFilterSensitiveLog = GetBlacklistReportsRequestFilterSensitiveLog;
const GetBlacklistReportsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetBlacklistReportsResponseFilterSensitiveLog = GetBlacklistReportsResponseFilterSensitiveLog;
const GetConfigurationSetRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetConfigurationSetRequestFilterSensitiveLog = GetConfigurationSetRequestFilterSensitiveLog;
const GetConfigurationSetResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetConfigurationSetResponseFilterSensitiveLog = GetConfigurationSetResponseFilterSensitiveLog;
const GetConfigurationSetEventDestinationsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetConfigurationSetEventDestinationsRequestFilterSensitiveLog = GetConfigurationSetEventDestinationsRequestFilterSensitiveLog;
const GetConfigurationSetEventDestinationsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetConfigurationSetEventDestinationsResponseFilterSensitiveLog = GetConfigurationSetEventDestinationsResponseFilterSensitiveLog;
const GetContactRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetContactRequestFilterSensitiveLog = GetContactRequestFilterSensitiveLog;
const GetContactResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetContactResponseFilterSensitiveLog = GetContactResponseFilterSensitiveLog;
const GetContactListRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetContactListRequestFilterSensitiveLog = GetContactListRequestFilterSensitiveLog;
const GetContactListResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetContactListResponseFilterSensitiveLog = GetContactListResponseFilterSensitiveLog;
const GetCustomVerificationEmailTemplateRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetCustomVerificationEmailTemplateRequestFilterSensitiveLog = GetCustomVerificationEmailTemplateRequestFilterSensitiveLog;
const GetCustomVerificationEmailTemplateResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetCustomVerificationEmailTemplateResponseFilterSensitiveLog = GetCustomVerificationEmailTemplateResponseFilterSensitiveLog;
const GetDedicatedIpRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDedicatedIpRequestFilterSensitiveLog = GetDedicatedIpRequestFilterSensitiveLog;
const GetDedicatedIpResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDedicatedIpResponseFilterSensitiveLog = GetDedicatedIpResponseFilterSensitiveLog;
const GetDedicatedIpsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDedicatedIpsRequestFilterSensitiveLog = GetDedicatedIpsRequestFilterSensitiveLog;
const GetDedicatedIpsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDedicatedIpsResponseFilterSensitiveLog = GetDedicatedIpsResponseFilterSensitiveLog;
const GetDeliverabilityDashboardOptionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDeliverabilityDashboardOptionsRequestFilterSensitiveLog = GetDeliverabilityDashboardOptionsRequestFilterSensitiveLog;
const GetDeliverabilityDashboardOptionsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDeliverabilityDashboardOptionsResponseFilterSensitiveLog = GetDeliverabilityDashboardOptionsResponseFilterSensitiveLog;
const GetDeliverabilityTestReportRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDeliverabilityTestReportRequestFilterSensitiveLog = GetDeliverabilityTestReportRequestFilterSensitiveLog;
const PlacementStatisticsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PlacementStatisticsFilterSensitiveLog = PlacementStatisticsFilterSensitiveLog;
const IspPlacementFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.IspPlacementFilterSensitiveLog = IspPlacementFilterSensitiveLog;
const GetDeliverabilityTestReportResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDeliverabilityTestReportResponseFilterSensitiveLog = GetDeliverabilityTestReportResponseFilterSensitiveLog;
const GetDomainDeliverabilityCampaignRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDomainDeliverabilityCampaignRequestFilterSensitiveLog = GetDomainDeliverabilityCampaignRequestFilterSensitiveLog;
const GetDomainDeliverabilityCampaignResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDomainDeliverabilityCampaignResponseFilterSensitiveLog = GetDomainDeliverabilityCampaignResponseFilterSensitiveLog;
const GetDomainStatisticsReportRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDomainStatisticsReportRequestFilterSensitiveLog = GetDomainStatisticsReportRequestFilterSensitiveLog;
const OverallVolumeFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.OverallVolumeFilterSensitiveLog = OverallVolumeFilterSensitiveLog;
const GetDomainStatisticsReportResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDomainStatisticsReportResponseFilterSensitiveLog = GetDomainStatisticsReportResponseFilterSensitiveLog;
const GetEmailIdentityRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetEmailIdentityRequestFilterSensitiveLog = GetEmailIdentityRequestFilterSensitiveLog;
const MailFromAttributesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.MailFromAttributesFilterSensitiveLog = MailFromAttributesFilterSensitiveLog;
const GetEmailIdentityResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetEmailIdentityResponseFilterSensitiveLog = GetEmailIdentityResponseFilterSensitiveLog;
const GetEmailIdentityPoliciesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetEmailIdentityPoliciesRequestFilterSensitiveLog = GetEmailIdentityPoliciesRequestFilterSensitiveLog;
const GetEmailIdentityPoliciesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetEmailIdentityPoliciesResponseFilterSensitiveLog = GetEmailIdentityPoliciesResponseFilterSensitiveLog;
const GetEmailTemplateRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetEmailTemplateRequestFilterSensitiveLog = GetEmailTemplateRequestFilterSensitiveLog;
const GetEmailTemplateResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetEmailTemplateResponseFilterSensitiveLog = GetEmailTemplateResponseFilterSensitiveLog;
const GetImportJobRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetImportJobRequestFilterSensitiveLog = GetImportJobRequestFilterSensitiveLog;
const GetImportJobResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetImportJobResponseFilterSensitiveLog = GetImportJobResponseFilterSensitiveLog;
const GetSuppressedDestinationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSuppressedDestinationRequestFilterSensitiveLog = GetSuppressedDestinationRequestFilterSensitiveLog;
const SuppressedDestinationAttributesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SuppressedDestinationAttributesFilterSensitiveLog = SuppressedDestinationAttributesFilterSensitiveLog;
const SuppressedDestinationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SuppressedDestinationFilterSensitiveLog = SuppressedDestinationFilterSensitiveLog;
const GetSuppressedDestinationResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetSuppressedDestinationResponseFilterSensitiveLog = GetSuppressedDestinationResponseFilterSensitiveLog;
const IdentityInfoFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.IdentityInfoFilterSensitiveLog = IdentityInfoFilterSensitiveLog;
const ImportJobSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ImportJobSummaryFilterSensitiveLog = ImportJobSummaryFilterSensitiveLog;
const ListConfigurationSetsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListConfigurationSetsRequestFilterSensitiveLog = ListConfigurationSetsRequestFilterSensitiveLog;
const ListConfigurationSetsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListConfigurationSetsResponseFilterSensitiveLog = ListConfigurationSetsResponseFilterSensitiveLog;
const ListContactListsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListContactListsRequestFilterSensitiveLog = ListContactListsRequestFilterSensitiveLog;
const ListContactListsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListContactListsResponseFilterSensitiveLog = ListContactListsResponseFilterSensitiveLog;
const TopicFilterFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TopicFilterFilterSensitiveLog = TopicFilterFilterSensitiveLog;
const ListContactsFilterFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListContactsFilterFilterSensitiveLog = ListContactsFilterFilterSensitiveLog;
const ListContactsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListContactsRequestFilterSensitiveLog = ListContactsRequestFilterSensitiveLog;
const ListContactsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListContactsResponseFilterSensitiveLog = ListContactsResponseFilterSensitiveLog;
const ListCustomVerificationEmailTemplatesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListCustomVerificationEmailTemplatesRequestFilterSensitiveLog = ListCustomVerificationEmailTemplatesRequestFilterSensitiveLog;
const ListCustomVerificationEmailTemplatesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListCustomVerificationEmailTemplatesResponseFilterSensitiveLog = ListCustomVerificationEmailTemplatesResponseFilterSensitiveLog;
const ListDedicatedIpPoolsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListDedicatedIpPoolsRequestFilterSensitiveLog = ListDedicatedIpPoolsRequestFilterSensitiveLog;
const ListDedicatedIpPoolsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListDedicatedIpPoolsResponseFilterSensitiveLog = ListDedicatedIpPoolsResponseFilterSensitiveLog;
const ListDeliverabilityTestReportsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListDeliverabilityTestReportsRequestFilterSensitiveLog = ListDeliverabilityTestReportsRequestFilterSensitiveLog;
const ListDeliverabilityTestReportsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListDeliverabilityTestReportsResponseFilterSensitiveLog = ListDeliverabilityTestReportsResponseFilterSensitiveLog;
const ListDomainDeliverabilityCampaignsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListDomainDeliverabilityCampaignsRequestFilterSensitiveLog = ListDomainDeliverabilityCampaignsRequestFilterSensitiveLog;
const ListDomainDeliverabilityCampaignsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListDomainDeliverabilityCampaignsResponseFilterSensitiveLog = ListDomainDeliverabilityCampaignsResponseFilterSensitiveLog;
const ListEmailIdentitiesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListEmailIdentitiesRequestFilterSensitiveLog = ListEmailIdentitiesRequestFilterSensitiveLog;
const ListEmailIdentitiesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListEmailIdentitiesResponseFilterSensitiveLog = ListEmailIdentitiesResponseFilterSensitiveLog;
const ListEmailTemplatesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListEmailTemplatesRequestFilterSensitiveLog = ListEmailTemplatesRequestFilterSensitiveLog;
const ListEmailTemplatesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListEmailTemplatesResponseFilterSensitiveLog = ListEmailTemplatesResponseFilterSensitiveLog;
const ListImportJobsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListImportJobsRequestFilterSensitiveLog = ListImportJobsRequestFilterSensitiveLog;
const ListImportJobsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListImportJobsResponseFilterSensitiveLog = ListImportJobsResponseFilterSensitiveLog;
const ListManagementOptionsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListManagementOptionsFilterSensitiveLog = ListManagementOptionsFilterSensitiveLog;
const ListSuppressedDestinationsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListSuppressedDestinationsRequestFilterSensitiveLog = ListSuppressedDestinationsRequestFilterSensitiveLog;
const SuppressedDestinationSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SuppressedDestinationSummaryFilterSensitiveLog = SuppressedDestinationSummaryFilterSensitiveLog;
const ListSuppressedDestinationsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListSuppressedDestinationsResponseFilterSensitiveLog = ListSuppressedDestinationsResponseFilterSensitiveLog;
const ListTagsForResourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListTagsForResourceRequestFilterSensitiveLog = ListTagsForResourceRequestFilterSensitiveLog;
const ListTagsForResourceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListTagsForResourceResponseFilterSensitiveLog = ListTagsForResourceResponseFilterSensitiveLog;
const PutAccountDedicatedIpWarmupAttributesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutAccountDedicatedIpWarmupAttributesRequestFilterSensitiveLog = PutAccountDedicatedIpWarmupAttributesRequestFilterSensitiveLog;
const PutAccountDedicatedIpWarmupAttributesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutAccountDedicatedIpWarmupAttributesResponseFilterSensitiveLog = PutAccountDedicatedIpWarmupAttributesResponseFilterSensitiveLog;
const PutAccountDetailsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.WebsiteURL && { WebsiteURL: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.UseCaseDescription && { UseCaseDescription: smithy_client_1.SENSITIVE_STRING }),
    ...(obj.AdditionalContactEmailAddresses && { AdditionalContactEmailAddresses: smithy_client_1.SENSITIVE_STRING }),
});
exports.PutAccountDetailsRequestFilterSensitiveLog = PutAccountDetailsRequestFilterSensitiveLog;
const PutAccountDetailsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutAccountDetailsResponseFilterSensitiveLog = PutAccountDetailsResponseFilterSensitiveLog;
const PutAccountSendingAttributesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutAccountSendingAttributesRequestFilterSensitiveLog = PutAccountSendingAttributesRequestFilterSensitiveLog;
const PutAccountSendingAttributesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutAccountSendingAttributesResponseFilterSensitiveLog = PutAccountSendingAttributesResponseFilterSensitiveLog;
const PutAccountSuppressionAttributesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutAccountSuppressionAttributesRequestFilterSensitiveLog = PutAccountSuppressionAttributesRequestFilterSensitiveLog;
const PutAccountSuppressionAttributesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutAccountSuppressionAttributesResponseFilterSensitiveLog = PutAccountSuppressionAttributesResponseFilterSensitiveLog;
const PutConfigurationSetDeliveryOptionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutConfigurationSetDeliveryOptionsRequestFilterSensitiveLog = PutConfigurationSetDeliveryOptionsRequestFilterSensitiveLog;
const PutConfigurationSetDeliveryOptionsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutConfigurationSetDeliveryOptionsResponseFilterSensitiveLog = PutConfigurationSetDeliveryOptionsResponseFilterSensitiveLog;
const PutConfigurationSetReputationOptionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutConfigurationSetReputationOptionsRequestFilterSensitiveLog = PutConfigurationSetReputationOptionsRequestFilterSensitiveLog;
const PutConfigurationSetReputationOptionsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutConfigurationSetReputationOptionsResponseFilterSensitiveLog = PutConfigurationSetReputationOptionsResponseFilterSensitiveLog;
const PutConfigurationSetSendingOptionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutConfigurationSetSendingOptionsRequestFilterSensitiveLog = PutConfigurationSetSendingOptionsRequestFilterSensitiveLog;
const PutConfigurationSetSendingOptionsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutConfigurationSetSendingOptionsResponseFilterSensitiveLog = PutConfigurationSetSendingOptionsResponseFilterSensitiveLog;
const PutConfigurationSetSuppressionOptionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutConfigurationSetSuppressionOptionsRequestFilterSensitiveLog = PutConfigurationSetSuppressionOptionsRequestFilterSensitiveLog;
const PutConfigurationSetSuppressionOptionsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutConfigurationSetSuppressionOptionsResponseFilterSensitiveLog = PutConfigurationSetSuppressionOptionsResponseFilterSensitiveLog;
const PutConfigurationSetTrackingOptionsRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutConfigurationSetTrackingOptionsRequestFilterSensitiveLog = PutConfigurationSetTrackingOptionsRequestFilterSensitiveLog;
const PutConfigurationSetTrackingOptionsResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutConfigurationSetTrackingOptionsResponseFilterSensitiveLog = PutConfigurationSetTrackingOptionsResponseFilterSensitiveLog;
const PutDedicatedIpInPoolRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutDedicatedIpInPoolRequestFilterSensitiveLog = PutDedicatedIpInPoolRequestFilterSensitiveLog;
const PutDedicatedIpInPoolResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutDedicatedIpInPoolResponseFilterSensitiveLog = PutDedicatedIpInPoolResponseFilterSensitiveLog;
const PutDedicatedIpWarmupAttributesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutDedicatedIpWarmupAttributesRequestFilterSensitiveLog = PutDedicatedIpWarmupAttributesRequestFilterSensitiveLog;
const PutDedicatedIpWarmupAttributesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutDedicatedIpWarmupAttributesResponseFilterSensitiveLog = PutDedicatedIpWarmupAttributesResponseFilterSensitiveLog;
const PutDeliverabilityDashboardOptionRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutDeliverabilityDashboardOptionRequestFilterSensitiveLog = PutDeliverabilityDashboardOptionRequestFilterSensitiveLog;
const PutDeliverabilityDashboardOptionResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutDeliverabilityDashboardOptionResponseFilterSensitiveLog = PutDeliverabilityDashboardOptionResponseFilterSensitiveLog;
const PutEmailIdentityConfigurationSetAttributesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutEmailIdentityConfigurationSetAttributesRequestFilterSensitiveLog = PutEmailIdentityConfigurationSetAttributesRequestFilterSensitiveLog;
const PutEmailIdentityConfigurationSetAttributesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutEmailIdentityConfigurationSetAttributesResponseFilterSensitiveLog = PutEmailIdentityConfigurationSetAttributesResponseFilterSensitiveLog;
const PutEmailIdentityDkimAttributesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutEmailIdentityDkimAttributesRequestFilterSensitiveLog = PutEmailIdentityDkimAttributesRequestFilterSensitiveLog;
const PutEmailIdentityDkimAttributesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutEmailIdentityDkimAttributesResponseFilterSensitiveLog = PutEmailIdentityDkimAttributesResponseFilterSensitiveLog;
const PutEmailIdentityDkimSigningAttributesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
    ...(obj.SigningAttributes && { SigningAttributes: (0, exports.DkimSigningAttributesFilterSensitiveLog)(obj.SigningAttributes) }),
});
exports.PutEmailIdentityDkimSigningAttributesRequestFilterSensitiveLog = PutEmailIdentityDkimSigningAttributesRequestFilterSensitiveLog;
const PutEmailIdentityDkimSigningAttributesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutEmailIdentityDkimSigningAttributesResponseFilterSensitiveLog = PutEmailIdentityDkimSigningAttributesResponseFilterSensitiveLog;
const PutEmailIdentityFeedbackAttributesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutEmailIdentityFeedbackAttributesRequestFilterSensitiveLog = PutEmailIdentityFeedbackAttributesRequestFilterSensitiveLog;
const PutEmailIdentityFeedbackAttributesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutEmailIdentityFeedbackAttributesResponseFilterSensitiveLog = PutEmailIdentityFeedbackAttributesResponseFilterSensitiveLog;
const PutEmailIdentityMailFromAttributesRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutEmailIdentityMailFromAttributesRequestFilterSensitiveLog = PutEmailIdentityMailFromAttributesRequestFilterSensitiveLog;
const PutEmailIdentityMailFromAttributesResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutEmailIdentityMailFromAttributesResponseFilterSensitiveLog = PutEmailIdentityMailFromAttributesResponseFilterSensitiveLog;
const PutSuppressedDestinationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutSuppressedDestinationRequestFilterSensitiveLog = PutSuppressedDestinationRequestFilterSensitiveLog;
const PutSuppressedDestinationResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PutSuppressedDestinationResponseFilterSensitiveLog = PutSuppressedDestinationResponseFilterSensitiveLog;
const SendBulkEmailRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SendBulkEmailRequestFilterSensitiveLog = SendBulkEmailRequestFilterSensitiveLog;
const SendBulkEmailResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SendBulkEmailResponseFilterSensitiveLog = SendBulkEmailResponseFilterSensitiveLog;
const SendCustomVerificationEmailRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SendCustomVerificationEmailRequestFilterSensitiveLog = SendCustomVerificationEmailRequestFilterSensitiveLog;
const SendCustomVerificationEmailResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SendCustomVerificationEmailResponseFilterSensitiveLog = SendCustomVerificationEmailResponseFilterSensitiveLog;
const SendEmailRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SendEmailRequestFilterSensitiveLog = SendEmailRequestFilterSensitiveLog;
const SendEmailResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.SendEmailResponseFilterSensitiveLog = SendEmailResponseFilterSensitiveLog;
const TagResourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TagResourceRequestFilterSensitiveLog = TagResourceRequestFilterSensitiveLog;
const TagResourceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TagResourceResponseFilterSensitiveLog = TagResourceResponseFilterSensitiveLog;
const TestRenderEmailTemplateRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TestRenderEmailTemplateRequestFilterSensitiveLog = TestRenderEmailTemplateRequestFilterSensitiveLog;
const TestRenderEmailTemplateResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TestRenderEmailTemplateResponseFilterSensitiveLog = TestRenderEmailTemplateResponseFilterSensitiveLog;
const UntagResourceRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UntagResourceRequestFilterSensitiveLog = UntagResourceRequestFilterSensitiveLog;
const UntagResourceResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UntagResourceResponseFilterSensitiveLog = UntagResourceResponseFilterSensitiveLog;
const UpdateConfigurationSetEventDestinationRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateConfigurationSetEventDestinationRequestFilterSensitiveLog = UpdateConfigurationSetEventDestinationRequestFilterSensitiveLog;
const UpdateConfigurationSetEventDestinationResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateConfigurationSetEventDestinationResponseFilterSensitiveLog = UpdateConfigurationSetEventDestinationResponseFilterSensitiveLog;
const UpdateContactRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateContactRequestFilterSensitiveLog = UpdateContactRequestFilterSensitiveLog;
const UpdateContactResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateContactResponseFilterSensitiveLog = UpdateContactResponseFilterSensitiveLog;
const UpdateContactListRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateContactListRequestFilterSensitiveLog = UpdateContactListRequestFilterSensitiveLog;
const UpdateContactListResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateContactListResponseFilterSensitiveLog = UpdateContactListResponseFilterSensitiveLog;
const UpdateCustomVerificationEmailTemplateRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateCustomVerificationEmailTemplateRequestFilterSensitiveLog = UpdateCustomVerificationEmailTemplateRequestFilterSensitiveLog;
const UpdateCustomVerificationEmailTemplateResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateCustomVerificationEmailTemplateResponseFilterSensitiveLog = UpdateCustomVerificationEmailTemplateResponseFilterSensitiveLog;
const UpdateEmailIdentityPolicyRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateEmailIdentityPolicyRequestFilterSensitiveLog = UpdateEmailIdentityPolicyRequestFilterSensitiveLog;
const UpdateEmailIdentityPolicyResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateEmailIdentityPolicyResponseFilterSensitiveLog = UpdateEmailIdentityPolicyResponseFilterSensitiveLog;
const UpdateEmailTemplateRequestFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateEmailTemplateRequestFilterSensitiveLog = UpdateEmailTemplateRequestFilterSensitiveLog;
const UpdateEmailTemplateResponseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateEmailTemplateResponseFilterSensitiveLog = UpdateEmailTemplateResponseFilterSensitiveLog;
