// @aws-sdk/client-sns/dist-es/commands/GetDataProtectionPolicyCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetDataProtectionPolicyInputFilterSensitiveLog, GetDataProtectionPolicyResponseFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_queryGetDataProtectionPolicyCommand, serializeAws_queryGetDataProtectionPolicyCommand, } from "../protocols/Aws_query";
var GetDataProtectionPolicyCommand = (function (_super) {
    __extends(GetDataProtectionPolicyCommand, _super);
    function GetDataProtectionPolicyCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetDataProtectionPolicyCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "SNSClient";
        var commandName = "GetDataProtectionPolicyCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetDataProtectionPolicyInputFilterSensitiveLog,
            outputFilterSensitiveLog: GetDataProtectionPolicyResponseFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetDataProtectionPolicyCommand.prototype.serialize = function (input, context) {
        return serializeAws_queryGetDataProtectionPolicyCommand(input, context);
    };
    GetDataProtectionPolicyCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_queryGetDataProtectionPolicyCommand(output, context);
    };
    return GetDataProtectionPolicyCommand;
}($Command));
export { GetDataProtectionPolicyCommand };
