// @aws-sdk/client-athena/dist-cjs/models/models_0.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DeleteWorkGroupInputFilterSensitiveLog = exports.DeletePreparedStatementOutputFilterSensitiveLog = exports.DeletePreparedStatementInputFilterSensitiveLog = exports.DeleteNamedQueryOutputFilterSensitiveLog = exports.DeleteNamedQueryInputFilterSensitiveLog = exports.DeleteDataCatalogOutputFilterSensitiveLog = exports.DeleteDataCatalogInputFilterSensitiveLog = exports.CreateWorkGroupOutputFilterSensitiveLog = exports.CreateWorkGroupInputFilterSensitiveLog = exports.WorkGroupConfigurationFilterSensitiveLog = exports.CreatePreparedStatementOutputFilterSensitiveLog = exports.CreatePreparedStatementInputFilterSensitiveLog = exports.CreateNamedQueryOutputFilterSensitiveLog = exports.CreateNamedQueryInputFilterSensitiveLog = exports.CreateDataCatalogOutputFilterSensitiveLog = exports.CreateDataCatalogInputFilterSensitiveLog = exports.TagFilterSensitiveLog = exports.BatchGetQueryExecutionOutputFilterSensitiveLog = exports.UnprocessedQueryExecutionIdFilterSensitiveLog = exports.QueryExecutionFilterSensitiveLog = exports.QueryExecutionStatusFilterSensitiveLog = exports.AthenaErrorFilterSensitiveLog = exports.QueryExecutionStatisticsFilterSensitiveLog = exports.ResultConfigurationFilterSensitiveLog = exports.EncryptionConfigurationFilterSensitiveLog = exports.QueryExecutionContextFilterSensitiveLog = exports.EngineVersionFilterSensitiveLog = exports.BatchGetQueryExecutionInputFilterSensitiveLog = exports.BatchGetPreparedStatementOutputFilterSensitiveLog = exports.UnprocessedPreparedStatementNameFilterSensitiveLog = exports.PreparedStatementFilterSensitiveLog = exports.BatchGetPreparedStatementInputFilterSensitiveLog = exports.BatchGetNamedQueryOutputFilterSensitiveLog = exports.UnprocessedNamedQueryIdFilterSensitiveLog = exports.NamedQueryFilterSensitiveLog = exports.BatchGetNamedQueryInputFilterSensitiveLog = exports.AclConfigurationFilterSensitiveLog = exports.TooManyRequestsException = exports.ThrottleReason = exports.WorkGroupState = exports.ColumnNullable = exports.MetadataException = exports.ResourceNotFoundException = exports.DataCatalogType = exports.QueryExecutionState = exports.StatementType = exports.EncryptionOption = exports.InvalidRequestException = exports.InternalServerException = exports.S3AclOption = void 0;
exports.WorkGroupSummaryFilterSensitiveLog = exports.ListWorkGroupsInputFilterSensitiveLog = exports.ListTagsForResourceOutputFilterSensitiveLog = exports.ListTagsForResourceInputFilterSensitiveLog = exports.ListTableMetadataOutputFilterSensitiveLog = exports.ListTableMetadataInputFilterSensitiveLog = exports.ListQueryExecutionsOutputFilterSensitiveLog = exports.ListQueryExecutionsInputFilterSensitiveLog = exports.ListPreparedStatementsOutputFilterSensitiveLog = exports.PreparedStatementSummaryFilterSensitiveLog = exports.ListPreparedStatementsInputFilterSensitiveLog = exports.ListNamedQueriesOutputFilterSensitiveLog = exports.ListNamedQueriesInputFilterSensitiveLog = exports.ListEngineVersionsOutputFilterSensitiveLog = exports.ListEngineVersionsInputFilterSensitiveLog = exports.ListDataCatalogsOutputFilterSensitiveLog = exports.DataCatalogSummaryFilterSensitiveLog = exports.ListDataCatalogsInputFilterSensitiveLog = exports.ListDatabasesOutputFilterSensitiveLog = exports.ListDatabasesInputFilterSensitiveLog = exports.GetWorkGroupOutputFilterSensitiveLog = exports.WorkGroupFilterSensitiveLog = exports.GetWorkGroupInputFilterSensitiveLog = exports.GetTableMetadataOutputFilterSensitiveLog = exports.TableMetadataFilterSensitiveLog = exports.ColumnFilterSensitiveLog = exports.GetTableMetadataInputFilterSensitiveLog = exports.QueryRuntimeStatisticsTimelineFilterSensitiveLog = exports.QueryRuntimeStatisticsRowsFilterSensitiveLog = exports.GetQueryRuntimeStatisticsInputFilterSensitiveLog = exports.GetQueryResultsOutputFilterSensitiveLog = exports.ResultSetFilterSensitiveLog = exports.RowFilterSensitiveLog = exports.DatumFilterSensitiveLog = exports.ResultSetMetadataFilterSensitiveLog = exports.ColumnInfoFilterSensitiveLog = exports.GetQueryResultsInputFilterSensitiveLog = exports.GetQueryExecutionOutputFilterSensitiveLog = exports.GetQueryExecutionInputFilterSensitiveLog = exports.GetPreparedStatementOutputFilterSensitiveLog = exports.GetPreparedStatementInputFilterSensitiveLog = exports.GetNamedQueryOutputFilterSensitiveLog = exports.GetNamedQueryInputFilterSensitiveLog = exports.GetDataCatalogOutputFilterSensitiveLog = exports.DataCatalogFilterSensitiveLog = exports.GetDataCatalogInputFilterSensitiveLog = exports.GetDatabaseOutputFilterSensitiveLog = exports.DatabaseFilterSensitiveLog = exports.GetDatabaseInputFilterSensitiveLog = exports.DeleteWorkGroupOutputFilterSensitiveLog = void 0;
exports.GetQueryRuntimeStatisticsOutputFilterSensitiveLog = exports.QueryRuntimeStatisticsFilterSensitiveLog = exports.QueryStageFilterSensitiveLog = exports.QueryStagePlanNodeFilterSensitiveLog = exports.UpdateWorkGroupOutputFilterSensitiveLog = exports.UpdateWorkGroupInputFilterSensitiveLog = exports.WorkGroupConfigurationUpdatesFilterSensitiveLog = exports.ResultConfigurationUpdatesFilterSensitiveLog = exports.UpdatePreparedStatementOutputFilterSensitiveLog = exports.UpdatePreparedStatementInputFilterSensitiveLog = exports.UpdateNamedQueryOutputFilterSensitiveLog = exports.UpdateNamedQueryInputFilterSensitiveLog = exports.UpdateDataCatalogOutputFilterSensitiveLog = exports.UpdateDataCatalogInputFilterSensitiveLog = exports.UntagResourceOutputFilterSensitiveLog = exports.UntagResourceInputFilterSensitiveLog = exports.TagResourceOutputFilterSensitiveLog = exports.TagResourceInputFilterSensitiveLog = exports.StopQueryExecutionOutputFilterSensitiveLog = exports.StopQueryExecutionInputFilterSensitiveLog = exports.StartQueryExecutionOutputFilterSensitiveLog = exports.StartQueryExecutionInputFilterSensitiveLog = exports.ListWorkGroupsOutputFilterSensitiveLog = void 0;
const AthenaServiceException_1 = require("./AthenaServiceException");
var S3AclOption;
(function (S3AclOption) {
    S3AclOption["BUCKET_OWNER_FULL_CONTROL"] = "BUCKET_OWNER_FULL_CONTROL";
})(S3AclOption = exports.S3AclOption || (exports.S3AclOption = {}));
class InternalServerException extends AthenaServiceException_1.AthenaServiceException {
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        this.name = "InternalServerException";
        this.$fault = "server";
        Object.setPrototypeOf(this, InternalServerException.prototype);
        this.Message = opts.Message;
    }
}
exports.InternalServerException = InternalServerException;
class InvalidRequestException extends AthenaServiceException_1.AthenaServiceException {
    constructor(opts) {
        super({
            name: "InvalidRequestException",
            $fault: "client",
            ...opts,
        });
        this.name = "InvalidRequestException";
        this.$fault = "client";
        Object.setPrototypeOf(this, InvalidRequestException.prototype);
        this.AthenaErrorCode = opts.AthenaErrorCode;
        this.Message = opts.Message;
    }
}
exports.InvalidRequestException = InvalidRequestException;
var EncryptionOption;
(function (EncryptionOption) {
    EncryptionOption["CSE_KMS"] = "CSE_KMS";
    EncryptionOption["SSE_KMS"] = "SSE_KMS";
    EncryptionOption["SSE_S3"] = "SSE_S3";
})(EncryptionOption = exports.EncryptionOption || (exports.EncryptionOption = {}));
var StatementType;
(function (StatementType) {
    StatementType["DDL"] = "DDL";
    StatementType["DML"] = "DML";
    StatementType["UTILITY"] = "UTILITY";
})(StatementType = exports.StatementType || (exports.StatementType = {}));
var QueryExecutionState;
(function (QueryExecutionState) {
    QueryExecutionState["CANCELLED"] = "CANCELLED";
    QueryExecutionState["FAILED"] = "FAILED";
    QueryExecutionState["QUEUED"] = "QUEUED";
    QueryExecutionState["RUNNING"] = "RUNNING";
    QueryExecutionState["SUCCEEDED"] = "SUCCEEDED";
})(QueryExecutionState = exports.QueryExecutionState || (exports.QueryExecutionState = {}));
var DataCatalogType;
(function (DataCatalogType) {
    DataCatalogType["GLUE"] = "GLUE";
    DataCatalogType["HIVE"] = "HIVE";
    DataCatalogType["LAMBDA"] = "LAMBDA";
})(DataCatalogType = exports.DataCatalogType || (exports.DataCatalogType = {}));
class ResourceNotFoundException extends AthenaServiceException_1.AthenaServiceException {
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        this.name = "ResourceNotFoundException";
        this.$fault = "client";
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
        this.Message = opts.Message;
        this.ResourceName = opts.ResourceName;
    }
}
exports.ResourceNotFoundException = ResourceNotFoundException;
class MetadataException extends AthenaServiceException_1.AthenaServiceException {
    constructor(opts) {
        super({
            name: "MetadataException",
            $fault: "client",
            ...opts,
        });
        this.name = "MetadataException";
        this.$fault = "client";
        Object.setPrototypeOf(this, MetadataException.prototype);
        this.Message = opts.Message;
    }
}
exports.MetadataException = MetadataException;
var ColumnNullable;
(function (ColumnNullable) {
    ColumnNullable["NOT_NULL"] = "NOT_NULL";
    ColumnNullable["NULLABLE"] = "NULLABLE";
    ColumnNullable["UNKNOWN"] = "UNKNOWN";
})(ColumnNullable = exports.ColumnNullable || (exports.ColumnNullable = {}));
var WorkGroupState;
(function (WorkGroupState) {
    WorkGroupState["DISABLED"] = "DISABLED";
    WorkGroupState["ENABLED"] = "ENABLED";
})(WorkGroupState = exports.WorkGroupState || (exports.WorkGroupState = {}));
var ThrottleReason;
(function (ThrottleReason) {
    ThrottleReason["CONCURRENT_QUERY_LIMIT_EXCEEDED"] = "CONCURRENT_QUERY_LIMIT_EXCEEDED";
})(ThrottleReason = exports.ThrottleReason || (exports.ThrottleReason = {}));
class TooManyRequestsException extends AthenaServiceException_1.AthenaServiceException {
    constructor(opts) {
        super({
            name: "TooManyRequestsException",
            $fault: "client",
            ...opts,
        });
        this.name = "TooManyRequestsException";
        this.$fault = "client";
        Object.setPrototypeOf(this, TooManyRequestsException.prototype);
        this.Message = opts.Message;
        this.Reason = opts.Reason;
    }
}
exports.TooManyRequestsException = TooManyRequestsException;
const AclConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.AclConfigurationFilterSensitiveLog = AclConfigurationFilterSensitiveLog;
const BatchGetNamedQueryInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BatchGetNamedQueryInputFilterSensitiveLog = BatchGetNamedQueryInputFilterSensitiveLog;
const NamedQueryFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.NamedQueryFilterSensitiveLog = NamedQueryFilterSensitiveLog;
const UnprocessedNamedQueryIdFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UnprocessedNamedQueryIdFilterSensitiveLog = UnprocessedNamedQueryIdFilterSensitiveLog;
const BatchGetNamedQueryOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BatchGetNamedQueryOutputFilterSensitiveLog = BatchGetNamedQueryOutputFilterSensitiveLog;
const BatchGetPreparedStatementInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BatchGetPreparedStatementInputFilterSensitiveLog = BatchGetPreparedStatementInputFilterSensitiveLog;
const PreparedStatementFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PreparedStatementFilterSensitiveLog = PreparedStatementFilterSensitiveLog;
const UnprocessedPreparedStatementNameFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UnprocessedPreparedStatementNameFilterSensitiveLog = UnprocessedPreparedStatementNameFilterSensitiveLog;
const BatchGetPreparedStatementOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BatchGetPreparedStatementOutputFilterSensitiveLog = BatchGetPreparedStatementOutputFilterSensitiveLog;
const BatchGetQueryExecutionInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BatchGetQueryExecutionInputFilterSensitiveLog = BatchGetQueryExecutionInputFilterSensitiveLog;
const EngineVersionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EngineVersionFilterSensitiveLog = EngineVersionFilterSensitiveLog;
const QueryExecutionContextFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.QueryExecutionContextFilterSensitiveLog = QueryExecutionContextFilterSensitiveLog;
const EncryptionConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.EncryptionConfigurationFilterSensitiveLog = EncryptionConfigurationFilterSensitiveLog;
const ResultConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ResultConfigurationFilterSensitiveLog = ResultConfigurationFilterSensitiveLog;
const QueryExecutionStatisticsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.QueryExecutionStatisticsFilterSensitiveLog = QueryExecutionStatisticsFilterSensitiveLog;
const AthenaErrorFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.AthenaErrorFilterSensitiveLog = AthenaErrorFilterSensitiveLog;
const QueryExecutionStatusFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.QueryExecutionStatusFilterSensitiveLog = QueryExecutionStatusFilterSensitiveLog;
const QueryExecutionFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.QueryExecutionFilterSensitiveLog = QueryExecutionFilterSensitiveLog;
const UnprocessedQueryExecutionIdFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UnprocessedQueryExecutionIdFilterSensitiveLog = UnprocessedQueryExecutionIdFilterSensitiveLog;
const BatchGetQueryExecutionOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.BatchGetQueryExecutionOutputFilterSensitiveLog = BatchGetQueryExecutionOutputFilterSensitiveLog;
const TagFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TagFilterSensitiveLog = TagFilterSensitiveLog;
const CreateDataCatalogInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateDataCatalogInputFilterSensitiveLog = CreateDataCatalogInputFilterSensitiveLog;
const CreateDataCatalogOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateDataCatalogOutputFilterSensitiveLog = CreateDataCatalogOutputFilterSensitiveLog;
const CreateNamedQueryInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateNamedQueryInputFilterSensitiveLog = CreateNamedQueryInputFilterSensitiveLog;
const CreateNamedQueryOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateNamedQueryOutputFilterSensitiveLog = CreateNamedQueryOutputFilterSensitiveLog;
const CreatePreparedStatementInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreatePreparedStatementInputFilterSensitiveLog = CreatePreparedStatementInputFilterSensitiveLog;
const CreatePreparedStatementOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreatePreparedStatementOutputFilterSensitiveLog = CreatePreparedStatementOutputFilterSensitiveLog;
const WorkGroupConfigurationFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.WorkGroupConfigurationFilterSensitiveLog = WorkGroupConfigurationFilterSensitiveLog;
const CreateWorkGroupInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateWorkGroupInputFilterSensitiveLog = CreateWorkGroupInputFilterSensitiveLog;
const CreateWorkGroupOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.CreateWorkGroupOutputFilterSensitiveLog = CreateWorkGroupOutputFilterSensitiveLog;
const DeleteDataCatalogInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteDataCatalogInputFilterSensitiveLog = DeleteDataCatalogInputFilterSensitiveLog;
const DeleteDataCatalogOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteDataCatalogOutputFilterSensitiveLog = DeleteDataCatalogOutputFilterSensitiveLog;
const DeleteNamedQueryInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteNamedQueryInputFilterSensitiveLog = DeleteNamedQueryInputFilterSensitiveLog;
const DeleteNamedQueryOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteNamedQueryOutputFilterSensitiveLog = DeleteNamedQueryOutputFilterSensitiveLog;
const DeletePreparedStatementInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeletePreparedStatementInputFilterSensitiveLog = DeletePreparedStatementInputFilterSensitiveLog;
const DeletePreparedStatementOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeletePreparedStatementOutputFilterSensitiveLog = DeletePreparedStatementOutputFilterSensitiveLog;
const DeleteWorkGroupInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteWorkGroupInputFilterSensitiveLog = DeleteWorkGroupInputFilterSensitiveLog;
const DeleteWorkGroupOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DeleteWorkGroupOutputFilterSensitiveLog = DeleteWorkGroupOutputFilterSensitiveLog;
const GetDatabaseInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDatabaseInputFilterSensitiveLog = GetDatabaseInputFilterSensitiveLog;
const DatabaseFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DatabaseFilterSensitiveLog = DatabaseFilterSensitiveLog;
const GetDatabaseOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDatabaseOutputFilterSensitiveLog = GetDatabaseOutputFilterSensitiveLog;
const GetDataCatalogInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDataCatalogInputFilterSensitiveLog = GetDataCatalogInputFilterSensitiveLog;
const DataCatalogFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DataCatalogFilterSensitiveLog = DataCatalogFilterSensitiveLog;
const GetDataCatalogOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetDataCatalogOutputFilterSensitiveLog = GetDataCatalogOutputFilterSensitiveLog;
const GetNamedQueryInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetNamedQueryInputFilterSensitiveLog = GetNamedQueryInputFilterSensitiveLog;
const GetNamedQueryOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetNamedQueryOutputFilterSensitiveLog = GetNamedQueryOutputFilterSensitiveLog;
const GetPreparedStatementInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetPreparedStatementInputFilterSensitiveLog = GetPreparedStatementInputFilterSensitiveLog;
const GetPreparedStatementOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetPreparedStatementOutputFilterSensitiveLog = GetPreparedStatementOutputFilterSensitiveLog;
const GetQueryExecutionInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetQueryExecutionInputFilterSensitiveLog = GetQueryExecutionInputFilterSensitiveLog;
const GetQueryExecutionOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetQueryExecutionOutputFilterSensitiveLog = GetQueryExecutionOutputFilterSensitiveLog;
const GetQueryResultsInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetQueryResultsInputFilterSensitiveLog = GetQueryResultsInputFilterSensitiveLog;
const ColumnInfoFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ColumnInfoFilterSensitiveLog = ColumnInfoFilterSensitiveLog;
const ResultSetMetadataFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ResultSetMetadataFilterSensitiveLog = ResultSetMetadataFilterSensitiveLog;
const DatumFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DatumFilterSensitiveLog = DatumFilterSensitiveLog;
const RowFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.RowFilterSensitiveLog = RowFilterSensitiveLog;
const ResultSetFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ResultSetFilterSensitiveLog = ResultSetFilterSensitiveLog;
const GetQueryResultsOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetQueryResultsOutputFilterSensitiveLog = GetQueryResultsOutputFilterSensitiveLog;
const GetQueryRuntimeStatisticsInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetQueryRuntimeStatisticsInputFilterSensitiveLog = GetQueryRuntimeStatisticsInputFilterSensitiveLog;
const QueryRuntimeStatisticsRowsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.QueryRuntimeStatisticsRowsFilterSensitiveLog = QueryRuntimeStatisticsRowsFilterSensitiveLog;
const QueryRuntimeStatisticsTimelineFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.QueryRuntimeStatisticsTimelineFilterSensitiveLog = QueryRuntimeStatisticsTimelineFilterSensitiveLog;
const GetTableMetadataInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetTableMetadataInputFilterSensitiveLog = GetTableMetadataInputFilterSensitiveLog;
const ColumnFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ColumnFilterSensitiveLog = ColumnFilterSensitiveLog;
const TableMetadataFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TableMetadataFilterSensitiveLog = TableMetadataFilterSensitiveLog;
const GetTableMetadataOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetTableMetadataOutputFilterSensitiveLog = GetTableMetadataOutputFilterSensitiveLog;
const GetWorkGroupInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetWorkGroupInputFilterSensitiveLog = GetWorkGroupInputFilterSensitiveLog;
const WorkGroupFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.WorkGroupFilterSensitiveLog = WorkGroupFilterSensitiveLog;
const GetWorkGroupOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetWorkGroupOutputFilterSensitiveLog = GetWorkGroupOutputFilterSensitiveLog;
const ListDatabasesInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListDatabasesInputFilterSensitiveLog = ListDatabasesInputFilterSensitiveLog;
const ListDatabasesOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListDatabasesOutputFilterSensitiveLog = ListDatabasesOutputFilterSensitiveLog;
const ListDataCatalogsInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListDataCatalogsInputFilterSensitiveLog = ListDataCatalogsInputFilterSensitiveLog;
const DataCatalogSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.DataCatalogSummaryFilterSensitiveLog = DataCatalogSummaryFilterSensitiveLog;
const ListDataCatalogsOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListDataCatalogsOutputFilterSensitiveLog = ListDataCatalogsOutputFilterSensitiveLog;
const ListEngineVersionsInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListEngineVersionsInputFilterSensitiveLog = ListEngineVersionsInputFilterSensitiveLog;
const ListEngineVersionsOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListEngineVersionsOutputFilterSensitiveLog = ListEngineVersionsOutputFilterSensitiveLog;
const ListNamedQueriesInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListNamedQueriesInputFilterSensitiveLog = ListNamedQueriesInputFilterSensitiveLog;
const ListNamedQueriesOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListNamedQueriesOutputFilterSensitiveLog = ListNamedQueriesOutputFilterSensitiveLog;
const ListPreparedStatementsInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListPreparedStatementsInputFilterSensitiveLog = ListPreparedStatementsInputFilterSensitiveLog;
const PreparedStatementSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.PreparedStatementSummaryFilterSensitiveLog = PreparedStatementSummaryFilterSensitiveLog;
const ListPreparedStatementsOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListPreparedStatementsOutputFilterSensitiveLog = ListPreparedStatementsOutputFilterSensitiveLog;
const ListQueryExecutionsInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListQueryExecutionsInputFilterSensitiveLog = ListQueryExecutionsInputFilterSensitiveLog;
const ListQueryExecutionsOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListQueryExecutionsOutputFilterSensitiveLog = ListQueryExecutionsOutputFilterSensitiveLog;
const ListTableMetadataInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListTableMetadataInputFilterSensitiveLog = ListTableMetadataInputFilterSensitiveLog;
const ListTableMetadataOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListTableMetadataOutputFilterSensitiveLog = ListTableMetadataOutputFilterSensitiveLog;
const ListTagsForResourceInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListTagsForResourceInputFilterSensitiveLog = ListTagsForResourceInputFilterSensitiveLog;
const ListTagsForResourceOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListTagsForResourceOutputFilterSensitiveLog = ListTagsForResourceOutputFilterSensitiveLog;
const ListWorkGroupsInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListWorkGroupsInputFilterSensitiveLog = ListWorkGroupsInputFilterSensitiveLog;
const WorkGroupSummaryFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.WorkGroupSummaryFilterSensitiveLog = WorkGroupSummaryFilterSensitiveLog;
const ListWorkGroupsOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ListWorkGroupsOutputFilterSensitiveLog = ListWorkGroupsOutputFilterSensitiveLog;
const StartQueryExecutionInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartQueryExecutionInputFilterSensitiveLog = StartQueryExecutionInputFilterSensitiveLog;
const StartQueryExecutionOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StartQueryExecutionOutputFilterSensitiveLog = StartQueryExecutionOutputFilterSensitiveLog;
const StopQueryExecutionInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StopQueryExecutionInputFilterSensitiveLog = StopQueryExecutionInputFilterSensitiveLog;
const StopQueryExecutionOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.StopQueryExecutionOutputFilterSensitiveLog = StopQueryExecutionOutputFilterSensitiveLog;
const TagResourceInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TagResourceInputFilterSensitiveLog = TagResourceInputFilterSensitiveLog;
const TagResourceOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.TagResourceOutputFilterSensitiveLog = TagResourceOutputFilterSensitiveLog;
const UntagResourceInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UntagResourceInputFilterSensitiveLog = UntagResourceInputFilterSensitiveLog;
const UntagResourceOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UntagResourceOutputFilterSensitiveLog = UntagResourceOutputFilterSensitiveLog;
const UpdateDataCatalogInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateDataCatalogInputFilterSensitiveLog = UpdateDataCatalogInputFilterSensitiveLog;
const UpdateDataCatalogOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateDataCatalogOutputFilterSensitiveLog = UpdateDataCatalogOutputFilterSensitiveLog;
const UpdateNamedQueryInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateNamedQueryInputFilterSensitiveLog = UpdateNamedQueryInputFilterSensitiveLog;
const UpdateNamedQueryOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateNamedQueryOutputFilterSensitiveLog = UpdateNamedQueryOutputFilterSensitiveLog;
const UpdatePreparedStatementInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdatePreparedStatementInputFilterSensitiveLog = UpdatePreparedStatementInputFilterSensitiveLog;
const UpdatePreparedStatementOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdatePreparedStatementOutputFilterSensitiveLog = UpdatePreparedStatementOutputFilterSensitiveLog;
const ResultConfigurationUpdatesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.ResultConfigurationUpdatesFilterSensitiveLog = ResultConfigurationUpdatesFilterSensitiveLog;
const WorkGroupConfigurationUpdatesFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.WorkGroupConfigurationUpdatesFilterSensitiveLog = WorkGroupConfigurationUpdatesFilterSensitiveLog;
const UpdateWorkGroupInputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateWorkGroupInputFilterSensitiveLog = UpdateWorkGroupInputFilterSensitiveLog;
const UpdateWorkGroupOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.UpdateWorkGroupOutputFilterSensitiveLog = UpdateWorkGroupOutputFilterSensitiveLog;
const QueryStagePlanNodeFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.QueryStagePlanNodeFilterSensitiveLog = QueryStagePlanNodeFilterSensitiveLog;
const QueryStageFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.QueryStageFilterSensitiveLog = QueryStageFilterSensitiveLog;
const QueryRuntimeStatisticsFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.QueryRuntimeStatisticsFilterSensitiveLog = QueryRuntimeStatisticsFilterSensitiveLog;
const GetQueryRuntimeStatisticsOutputFilterSensitiveLog = (obj) => ({
    ...obj,
});
exports.GetQueryRuntimeStatisticsOutputFilterSensitiveLog = GetQueryRuntimeStatisticsOutputFilterSensitiveLog;
