// @aws-sdk/client-athena/dist-es/commands/BatchGetNamedQueryCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { BatchGetNamedQueryInputFilterSensitiveLog, BatchGetNamedQueryOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1BatchGetNamedQueryCommand, serializeAws_json1_1BatchGetNamedQueryCommand, } from "../protocols/Aws_json1_1";
var BatchGetNamedQueryCommand = (function (_super) {
    __extends(BatchGetNamedQueryCommand, _super);
    function BatchGetNamedQueryCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    BatchGetNamedQueryCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "BatchGetNamedQueryCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: BatchGetNamedQueryInputFilterSensitiveLog,
            outputFilterSensitiveLog: BatchGetNamedQueryOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    BatchGetNamedQueryCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1BatchGetNamedQueryCommand(input, context);
    };
    BatchGetNamedQueryCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1BatchGetNamedQueryCommand(output, context);
    };
    return BatchGetNamedQueryCommand;
}($Command));
export { BatchGetNamedQueryCommand };
