// @aws-sdk/client-athena/dist-es/commands/BatchGetPreparedStatementCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { BatchGetPreparedStatementInputFilterSensitiveLog, BatchGetPreparedStatementOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1BatchGetPreparedStatementCommand, serializeAws_json1_1BatchGetPreparedStatementCommand, } from "../protocols/Aws_json1_1";
var BatchGetPreparedStatementCommand = (function (_super) {
    __extends(BatchGetPreparedStatementCommand, _super);
    function BatchGetPreparedStatementCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    BatchGetPreparedStatementCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "BatchGetPreparedStatementCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: BatchGetPreparedStatementInputFilterSensitiveLog,
            outputFilterSensitiveLog: BatchGetPreparedStatementOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    BatchGetPreparedStatementCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1BatchGetPreparedStatementCommand(input, context);
    };
    BatchGetPreparedStatementCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1BatchGetPreparedStatementCommand(output, context);
    };
    return BatchGetPreparedStatementCommand;
}($Command));
export { BatchGetPreparedStatementCommand };
