// @aws-sdk/client-athena/dist-es/commands/BatchGetQueryExecutionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { BatchGetQueryExecutionInputFilterSensitiveLog, BatchGetQueryExecutionOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1BatchGetQueryExecutionCommand, serializeAws_json1_1BatchGetQueryExecutionCommand, } from "../protocols/Aws_json1_1";
var BatchGetQueryExecutionCommand = (function (_super) {
    __extends(BatchGetQueryExecutionCommand, _super);
    function BatchGetQueryExecutionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    BatchGetQueryExecutionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "BatchGetQueryExecutionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: BatchGetQueryExecutionInputFilterSensitiveLog,
            outputFilterSensitiveLog: BatchGetQueryExecutionOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    BatchGetQueryExecutionCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1BatchGetQueryExecutionCommand(input, context);
    };
    BatchGetQueryExecutionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1BatchGetQueryExecutionCommand(output, context);
    };
    return BatchGetQueryExecutionCommand;
}($Command));
export { BatchGetQueryExecutionCommand };
