// @aws-sdk/client-athena/dist-es/commands/CreateDataCatalogCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateDataCatalogInputFilterSensitiveLog, CreateDataCatalogOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1CreateDataCatalogCommand, serializeAws_json1_1CreateDataCatalogCommand, } from "../protocols/Aws_json1_1";
var CreateDataCatalogCommand = (function (_super) {
    __extends(CreateDataCatalogCommand, _super);
    function CreateDataCatalogCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateDataCatalogCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "CreateDataCatalogCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateDataCatalogInputFilterSensitiveLog,
            outputFilterSensitiveLog: CreateDataCatalogOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateDataCatalogCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1CreateDataCatalogCommand(input, context);
    };
    CreateDataCatalogCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1CreateDataCatalogCommand(output, context);
    };
    return CreateDataCatalogCommand;
}($Command));
export { CreateDataCatalogCommand };
