// @aws-sdk/client-athena/dist-es/commands/CreatePreparedStatementCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreatePreparedStatementInputFilterSensitiveLog, CreatePreparedStatementOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1CreatePreparedStatementCommand, serializeAws_json1_1CreatePreparedStatementCommand, } from "../protocols/Aws_json1_1";
var CreatePreparedStatementCommand = (function (_super) {
    __extends(CreatePreparedStatementCommand, _super);
    function CreatePreparedStatementCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreatePreparedStatementCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "CreatePreparedStatementCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreatePreparedStatementInputFilterSensitiveLog,
            outputFilterSensitiveLog: CreatePreparedStatementOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreatePreparedStatementCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1CreatePreparedStatementCommand(input, context);
    };
    CreatePreparedStatementCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1CreatePreparedStatementCommand(output, context);
    };
    return CreatePreparedStatementCommand;
}($Command));
export { CreatePreparedStatementCommand };
