// @aws-sdk/client-athena/dist-es/commands/CreateNamedQueryCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { CreateNamedQueryInputFilterSensitiveLog, CreateNamedQueryOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1CreateNamedQueryCommand, serializeAws_json1_1CreateNamedQueryCommand, } from "../protocols/Aws_json1_1";
var CreateNamedQueryCommand = (function (_super) {
    __extends(CreateNamedQueryCommand, _super);
    function CreateNamedQueryCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    CreateNamedQueryCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "CreateNamedQueryCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: CreateNamedQueryInputFilterSensitiveLog,
            outputFilterSensitiveLog: CreateNamedQueryOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    CreateNamedQueryCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1CreateNamedQueryCommand(input, context);
    };
    CreateNamedQueryCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1CreateNamedQueryCommand(output, context);
    };
    return CreateNamedQueryCommand;
}($Command));
export { CreateNamedQueryCommand };
