// @aws-sdk/client-athena/dist-es/commands/DeleteDataCatalogCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteDataCatalogInputFilterSensitiveLog, DeleteDataCatalogOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1DeleteDataCatalogCommand, serializeAws_json1_1DeleteDataCatalogCommand, } from "../protocols/Aws_json1_1";
var DeleteDataCatalogCommand = (function (_super) {
    __extends(DeleteDataCatalogCommand, _super);
    function DeleteDataCatalogCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteDataCatalogCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "DeleteDataCatalogCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteDataCatalogInputFilterSensitiveLog,
            outputFilterSensitiveLog: DeleteDataCatalogOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteDataCatalogCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DeleteDataCatalogCommand(input, context);
    };
    DeleteDataCatalogCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DeleteDataCatalogCommand(output, context);
    };
    return DeleteDataCatalogCommand;
}($Command));
export { DeleteDataCatalogCommand };
