// @aws-sdk/client-athena/dist-es/commands/DeleteWorkGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { DeleteWorkGroupInputFilterSensitiveLog, DeleteWorkGroupOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1DeleteWorkGroupCommand, serializeAws_json1_1DeleteWorkGroupCommand, } from "../protocols/Aws_json1_1";
var DeleteWorkGroupCommand = (function (_super) {
    __extends(DeleteWorkGroupCommand, _super);
    function DeleteWorkGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    DeleteWorkGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "DeleteWorkGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: DeleteWorkGroupInputFilterSensitiveLog,
            outputFilterSensitiveLog: DeleteWorkGroupOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    DeleteWorkGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1DeleteWorkGroupCommand(input, context);
    };
    DeleteWorkGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1DeleteWorkGroupCommand(output, context);
    };
    return DeleteWorkGroupCommand;
}($Command));
export { DeleteWorkGroupCommand };
