// @aws-sdk/client-athena/dist-es/commands/GetDatabaseCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetDatabaseInputFilterSensitiveLog, GetDatabaseOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1GetDatabaseCommand, serializeAws_json1_1GetDatabaseCommand, } from "../protocols/Aws_json1_1";
var GetDatabaseCommand = (function (_super) {
    __extends(GetDatabaseCommand, _super);
    function GetDatabaseCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetDatabaseCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "GetDatabaseCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetDatabaseInputFilterSensitiveLog,
            outputFilterSensitiveLog: GetDatabaseOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetDatabaseCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetDatabaseCommand(input, context);
    };
    GetDatabaseCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetDatabaseCommand(output, context);
    };
    return GetDatabaseCommand;
}($Command));
export { GetDatabaseCommand };
