// @aws-sdk/client-athena/dist-es/commands/GetPreparedStatementCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetPreparedStatementInputFilterSensitiveLog, GetPreparedStatementOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1GetPreparedStatementCommand, serializeAws_json1_1GetPreparedStatementCommand, } from "../protocols/Aws_json1_1";
var GetPreparedStatementCommand = (function (_super) {
    __extends(GetPreparedStatementCommand, _super);
    function GetPreparedStatementCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetPreparedStatementCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "GetPreparedStatementCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetPreparedStatementInputFilterSensitiveLog,
            outputFilterSensitiveLog: GetPreparedStatementOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetPreparedStatementCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetPreparedStatementCommand(input, context);
    };
    GetPreparedStatementCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetPreparedStatementCommand(output, context);
    };
    return GetPreparedStatementCommand;
}($Command));
export { GetPreparedStatementCommand };
