// @aws-sdk/client-athena/dist-es/commands/GetQueryRuntimeStatisticsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetQueryRuntimeStatisticsInputFilterSensitiveLog, GetQueryRuntimeStatisticsOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1GetQueryRuntimeStatisticsCommand, serializeAws_json1_1GetQueryRuntimeStatisticsCommand, } from "../protocols/Aws_json1_1";
var GetQueryRuntimeStatisticsCommand = (function (_super) {
    __extends(GetQueryRuntimeStatisticsCommand, _super);
    function GetQueryRuntimeStatisticsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetQueryRuntimeStatisticsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "GetQueryRuntimeStatisticsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetQueryRuntimeStatisticsInputFilterSensitiveLog,
            outputFilterSensitiveLog: GetQueryRuntimeStatisticsOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetQueryRuntimeStatisticsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetQueryRuntimeStatisticsCommand(input, context);
    };
    GetQueryRuntimeStatisticsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetQueryRuntimeStatisticsCommand(output, context);
    };
    return GetQueryRuntimeStatisticsCommand;
}($Command));
export { GetQueryRuntimeStatisticsCommand };
