// @aws-sdk/client-athena/dist-es/commands/GetQueryExecutionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetQueryExecutionInputFilterSensitiveLog, GetQueryExecutionOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1GetQueryExecutionCommand, serializeAws_json1_1GetQueryExecutionCommand, } from "../protocols/Aws_json1_1";
var GetQueryExecutionCommand = (function (_super) {
    __extends(GetQueryExecutionCommand, _super);
    function GetQueryExecutionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetQueryExecutionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "GetQueryExecutionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetQueryExecutionInputFilterSensitiveLog,
            outputFilterSensitiveLog: GetQueryExecutionOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetQueryExecutionCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetQueryExecutionCommand(input, context);
    };
    GetQueryExecutionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetQueryExecutionCommand(output, context);
    };
    return GetQueryExecutionCommand;
}($Command));
export { GetQueryExecutionCommand };
