// @aws-sdk/client-athena/dist-es/commands/GetTableMetadataCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { GetTableMetadataInputFilterSensitiveLog, GetTableMetadataOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1GetTableMetadataCommand, serializeAws_json1_1GetTableMetadataCommand, } from "../protocols/Aws_json1_1";
var GetTableMetadataCommand = (function (_super) {
    __extends(GetTableMetadataCommand, _super);
    function GetTableMetadataCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    GetTableMetadataCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "GetTableMetadataCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: GetTableMetadataInputFilterSensitiveLog,
            outputFilterSensitiveLog: GetTableMetadataOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    GetTableMetadataCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1GetTableMetadataCommand(input, context);
    };
    GetTableMetadataCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1GetTableMetadataCommand(output, context);
    };
    return GetTableMetadataCommand;
}($Command));
export { GetTableMetadataCommand };
