// @aws-sdk/client-athena/dist-es/commands/ListDatabasesCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListDatabasesInputFilterSensitiveLog, ListDatabasesOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1ListDatabasesCommand, serializeAws_json1_1ListDatabasesCommand, } from "../protocols/Aws_json1_1";
var ListDatabasesCommand = (function (_super) {
    __extends(ListDatabasesCommand, _super);
    function ListDatabasesCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListDatabasesCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "ListDatabasesCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListDatabasesInputFilterSensitiveLog,
            outputFilterSensitiveLog: ListDatabasesOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListDatabasesCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListDatabasesCommand(input, context);
    };
    ListDatabasesCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListDatabasesCommand(output, context);
    };
    return ListDatabasesCommand;
}($Command));
export { ListDatabasesCommand };
