// @aws-sdk/client-athena/dist-es/commands/ListTableMetadataCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListTableMetadataInputFilterSensitiveLog, ListTableMetadataOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1ListTableMetadataCommand, serializeAws_json1_1ListTableMetadataCommand, } from "../protocols/Aws_json1_1";
var ListTableMetadataCommand = (function (_super) {
    __extends(ListTableMetadataCommand, _super);
    function ListTableMetadataCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListTableMetadataCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "ListTableMetadataCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListTableMetadataInputFilterSensitiveLog,
            outputFilterSensitiveLog: ListTableMetadataOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListTableMetadataCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListTableMetadataCommand(input, context);
    };
    ListTableMetadataCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListTableMetadataCommand(output, context);
    };
    return ListTableMetadataCommand;
}($Command));
export { ListTableMetadataCommand };
