// @aws-sdk/client-athena/dist-es/commands/ListQueryExecutionsCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { ListQueryExecutionsInputFilterSensitiveLog, ListQueryExecutionsOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1ListQueryExecutionsCommand, serializeAws_json1_1ListQueryExecutionsCommand, } from "../protocols/Aws_json1_1";
var ListQueryExecutionsCommand = (function (_super) {
    __extends(ListQueryExecutionsCommand, _super);
    function ListQueryExecutionsCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    ListQueryExecutionsCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "ListQueryExecutionsCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: ListQueryExecutionsInputFilterSensitiveLog,
            outputFilterSensitiveLog: ListQueryExecutionsOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    ListQueryExecutionsCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1ListQueryExecutionsCommand(input, context);
    };
    ListQueryExecutionsCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1ListQueryExecutionsCommand(output, context);
    };
    return ListQueryExecutionsCommand;
}($Command));
export { ListQueryExecutionsCommand };
