// @aws-sdk/client-athena/dist-es/commands/StartQueryExecutionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StartQueryExecutionInputFilterSensitiveLog, StartQueryExecutionOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1StartQueryExecutionCommand, serializeAws_json1_1StartQueryExecutionCommand, } from "../protocols/Aws_json1_1";
var StartQueryExecutionCommand = (function (_super) {
    __extends(StartQueryExecutionCommand, _super);
    function StartQueryExecutionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StartQueryExecutionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "StartQueryExecutionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StartQueryExecutionInputFilterSensitiveLog,
            outputFilterSensitiveLog: StartQueryExecutionOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StartQueryExecutionCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1StartQueryExecutionCommand(input, context);
    };
    StartQueryExecutionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1StartQueryExecutionCommand(output, context);
    };
    return StartQueryExecutionCommand;
}($Command));
export { StartQueryExecutionCommand };
