// @aws-sdk/client-athena/dist-es/commands/StopQueryExecutionCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { StopQueryExecutionInputFilterSensitiveLog, StopQueryExecutionOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1StopQueryExecutionCommand, serializeAws_json1_1StopQueryExecutionCommand, } from "../protocols/Aws_json1_1";
var StopQueryExecutionCommand = (function (_super) {
    __extends(StopQueryExecutionCommand, _super);
    function StopQueryExecutionCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    StopQueryExecutionCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "StopQueryExecutionCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: StopQueryExecutionInputFilterSensitiveLog,
            outputFilterSensitiveLog: StopQueryExecutionOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    StopQueryExecutionCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1StopQueryExecutionCommand(input, context);
    };
    StopQueryExecutionCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1StopQueryExecutionCommand(output, context);
    };
    return StopQueryExecutionCommand;
}($Command));
export { StopQueryExecutionCommand };
