// @aws-sdk/client-athena/dist-es/commands/UpdateWorkGroupCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateWorkGroupInputFilterSensitiveLog, UpdateWorkGroupOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1UpdateWorkGroupCommand, serializeAws_json1_1UpdateWorkGroupCommand, } from "../protocols/Aws_json1_1";
var UpdateWorkGroupCommand = (function (_super) {
    __extends(UpdateWorkGroupCommand, _super);
    function UpdateWorkGroupCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateWorkGroupCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "UpdateWorkGroupCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateWorkGroupInputFilterSensitiveLog,
            outputFilterSensitiveLog: UpdateWorkGroupOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateWorkGroupCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1UpdateWorkGroupCommand(input, context);
    };
    UpdateWorkGroupCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1UpdateWorkGroupCommand(output, context);
    };
    return UpdateWorkGroupCommand;
}($Command));
export { UpdateWorkGroupCommand };
