// @aws-sdk/client-athena/dist-es/commands/UpdatePreparedStatementCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdatePreparedStatementInputFilterSensitiveLog, UpdatePreparedStatementOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1UpdatePreparedStatementCommand, serializeAws_json1_1UpdatePreparedStatementCommand, } from "../protocols/Aws_json1_1";
var UpdatePreparedStatementCommand = (function (_super) {
    __extends(UpdatePreparedStatementCommand, _super);
    function UpdatePreparedStatementCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdatePreparedStatementCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "UpdatePreparedStatementCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdatePreparedStatementInputFilterSensitiveLog,
            outputFilterSensitiveLog: UpdatePreparedStatementOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdatePreparedStatementCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1UpdatePreparedStatementCommand(input, context);
    };
    UpdatePreparedStatementCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1UpdatePreparedStatementCommand(output, context);
    };
    return UpdatePreparedStatementCommand;
}($Command));
export { UpdatePreparedStatementCommand };
