// @aws-sdk/client-athena/dist-es/commands/UpdateNamedQueryCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateNamedQueryInputFilterSensitiveLog, UpdateNamedQueryOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1UpdateNamedQueryCommand, serializeAws_json1_1UpdateNamedQueryCommand, } from "../protocols/Aws_json1_1";
var UpdateNamedQueryCommand = (function (_super) {
    __extends(UpdateNamedQueryCommand, _super);
    function UpdateNamedQueryCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateNamedQueryCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "UpdateNamedQueryCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateNamedQueryInputFilterSensitiveLog,
            outputFilterSensitiveLog: UpdateNamedQueryOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateNamedQueryCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1UpdateNamedQueryCommand(input, context);
    };
    UpdateNamedQueryCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1UpdateNamedQueryCommand(output, context);
    };
    return UpdateNamedQueryCommand;
}($Command));
export { UpdateNamedQueryCommand };
