// @aws-sdk/client-athena/dist-es/commands/UpdateDataCatalogCommand.js
import { __extends } from "tslib";
import { getSerdePlugin } from "@aws-sdk/middleware-serde";
import { Command as $Command } from "@aws-sdk/smithy-client";
import { UpdateDataCatalogInputFilterSensitiveLog, UpdateDataCatalogOutputFilterSensitiveLog, } from "../models/models_0";
import { deserializeAws_json1_1UpdateDataCatalogCommand, serializeAws_json1_1UpdateDataCatalogCommand, } from "../protocols/Aws_json1_1";
var UpdateDataCatalogCommand = (function (_super) {
    __extends(UpdateDataCatalogCommand, _super);
    function UpdateDataCatalogCommand(input) {
        var _this = _super.call(this) || this;
        _this.input = input;
        return _this;
    }
    UpdateDataCatalogCommand.prototype.resolveMiddleware = function (clientStack, configuration, options) {
        this.middlewareStack.use(getSerdePlugin(configuration, this.serialize, this.deserialize));
        var stack = clientStack.concat(this.middlewareStack);
        var logger = configuration.logger;
        var clientName = "AthenaClient";
        var commandName = "UpdateDataCatalogCommand";
        var handlerExecutionContext = {
            logger: logger,
            clientName: clientName,
            commandName: commandName,
            inputFilterSensitiveLog: UpdateDataCatalogInputFilterSensitiveLog,
            outputFilterSensitiveLog: UpdateDataCatalogOutputFilterSensitiveLog,
        };
        var requestHandler = configuration.requestHandler;
        return stack.resolve(function (request) {
            return requestHandler.handle(request.request, options || {});
        }, handlerExecutionContext);
    };
    UpdateDataCatalogCommand.prototype.serialize = function (input, context) {
        return serializeAws_json1_1UpdateDataCatalogCommand(input, context);
    };
    UpdateDataCatalogCommand.prototype.deserialize = function (output, context) {
        return deserializeAws_json1_1UpdateDataCatalogCommand(output, context);
    };
    return UpdateDataCatalogCommand;
}($Command));
export { UpdateDataCatalogCommand };
